diff --git a/node_modules/@bivoglobal/payment-react-native/lib/module/components/BivoTextInput.js b/node_modules/@bivoglobal/payment-react-native/lib/module/components/BivoTextInput.js
index a224cfe..252c65e 100644
--- a/node_modules/@bivoglobal/payment-react-native/lib/module/components/BivoTextInput.js
+++ b/node_modules/@bivoglobal/payment-react-native/lib/module/components/BivoTextInput.js
@@ -18,7 +18,8 @@ export const BivoTextInput = ({
   errorMsg,
   defaultError = true,
   containerStyle,
-  textStyle
+  textStyle,
+  keyboardType
 }) => {
   const [value, setValue] = useState('');
   const lastErrorRef = useRef('');
@@ -68,6 +69,7 @@ export const BivoTextInput = ({
       onBlur: handleBlur,
       onFocus: handleFocus,
       placeholder: placeholder,
+      keyboardType: keyboardType,
       style: [styles.input, textStyle, showError ? {
         borderColor: 'red'
       } : {}]
diff --git a/node_modules/@bivoglobal/payment-react-native/lib/typescript/src/components/BivoTextInput.d.ts b/node_modules/@bivoglobal/payment-react-native/lib/typescript/src/components/BivoTextInput.d.ts
index e84a128..6739a68 100644
--- a/node_modules/@bivoglobal/payment-react-native/lib/typescript/src/components/BivoTextInput.d.ts
+++ b/node_modules/@bivoglobal/payment-react-native/lib/typescript/src/components/BivoTextInput.d.ts
@@ -1,5 +1,5 @@
 import React from 'react';
-import type { ViewStyle, TextStyle } from 'react-native';
+import type { KeyboardTypeOptions, ViewStyle, TextStyle } from 'react-native';
 import type { BivoSecureStore } from '../core/BivoSecureStore';
 interface Props {
     bivoStore: BivoSecureStore;
@@ -18,6 +18,7 @@ interface Props {
     defaultError?: boolean;
     containerStyle?: ViewStyle;
     textStyle?: TextStyle;
+    keyboardType?: KeyboardTypeOptions;
 }
 export declare const BivoTextInput: React.FC<Props>;
 export {};
diff --git a/node_modules/@bivoglobal/payment-react-native/src/components/BivoTextInput.tsx b/node_modules/@bivoglobal/payment-react-native/src/components/BivoTextInput.tsx
index 447d105..ade1df0 100644
--- a/node_modules/@bivoglobal/payment-react-native/src/components/BivoTextInput.tsx
+++ b/node_modules/@bivoglobal/payment-react-native/src/components/BivoTextInput.tsx
@@ -1,6 +1,6 @@
 import React, { useRef, useState } from 'react';
 import { View, TextInput, Text, StyleSheet, Image } from 'react-native';
-import type { ViewStyle, TextStyle } from 'react-native';
+import type { KeyboardTypeOptions, ViewStyle, TextStyle } from 'react-native';
 import type { BivoSecureStore } from '../core/BivoSecureStore';
 import { Icons } from '../assets';
 import { formatExpiry } from '../utils/utils';
@@ -19,6 +19,7 @@ interface Props {
   defaultError?: boolean;
   containerStyle?: ViewStyle;
   textStyle?: TextStyle;
+  keyboardType?: KeyboardTypeOptions;
 }
 
 export const BivoTextInput: React.FC<Props> = ({
@@ -35,6 +36,7 @@ export const BivoTextInput: React.FC<Props> = ({
   defaultError = true,
   containerStyle,
   textStyle,
+  keyboardType,
 }) => {
   const [value, setValue] = useState('');
   const lastErrorRef = useRef('');
@@ -100,6 +102,7 @@ export const BivoTextInput: React.FC<Props> = ({
         onBlur={handleBlur}
         onFocus={handleFocus}
         placeholder={placeholder}
+        keyboardType={keyboardType}
         style={[
           styles.input,
           textStyle,
