diff --git a/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift b/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
index a8dee30..26b1d32 100644
--- a/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
+++ b/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
@@ -65,12 +65,17 @@ final class FasterImageView: UIView {
         super.init(frame: .zero)
         addSubview(lazyImageView)
         lazyImageView.translatesAutoresizingMaskIntoConstraints = false
+
+        let bottomAnchor = lazyImageView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        bottomConstraint = bottomAnchor
+
         NSLayoutConstraint.activate([
             lazyImageView.topAnchor.constraint(equalTo: topAnchor),
-            lazyImageView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            bottomAnchor,
             lazyImageView.leadingAnchor.constraint(equalTo: leadingAnchor),
             lazyImageView.trailingAnchor.constraint(equalTo: trailingAnchor),
         ])
+        
         lazyImageView.pipeline = .shared
         lazyImageView.priority = .high
         lazyImageView.onCompletion = { [weak self] result in
@@ -86,6 +91,9 @@ final class FasterImageView: UIView {
 
     private lazy var lazyImageView = LazyImageView()
   
+    private var aspectRatioConstraint: NSLayoutConstraint?
+    private var bottomConstraint: NSLayoutConstraint?
+
     // MARK: - Callbacks
 
     @objc var onError: RCTDirectEventBlock?
@@ -233,20 +241,29 @@ final class FasterImageView: UIView {
         }
     }
 
-    var showActivityIndicator = false {
-        didSet {
-            let activity = UIActivityIndicatorView()
-            if self.activityColor != nil {
-                activity.color = self.activityColor
-            }
-            lazyImageView.placeholderView = activity
-        }
-    }
+    var showActivityIndicator = false
 
-    var activityColor: UIColor?
+    var activityColor: UIColor? = Color.clear
   
     var resizeMode = "contain" {
         didSet {
+            if (resizeMode == "topContain") {
+                lazyImageView.imageView.contentMode = .scaleAspectFit
+                bottomConstraint?.priority = .defaultLow
+
+                if aspectRatioConstraint == nil, let image = lazyImageView.imageView.image {
+                    aspectRatioConstraint?.isActive = false
+                    aspectRatioConstraint = lazyImageView.imageView.widthAnchor.constraint(
+                        equalTo: lazyImageView.imageView.heightAnchor,
+                        multiplier: image.size.width / image.size.height
+                    )
+                    aspectRatioConstraint?.isActive = true
+                }
+                return
+            }
+
+            bottomConstraint?.priority = .required
+            aspectRatioConstraint = nil
             let mode = ResizeMode(rawValue: resizeMode)
             lazyImageView.imageView.contentMode = mode?.contentMode ?? .scaleAspectFit
         }
@@ -362,11 +379,19 @@ fileprivate extension FasterImageView {
     func completionHandler(with result: Result<ImageResponse, Error>) {
         switch result {
         case .success(let value):
+            if resizeMode == "topContain" {
+                aspectRatioConstraint = lazyImageView.imageView.widthAnchor.constraint(
+                    equalTo: lazyImageView.imageView.heightAnchor,
+                    multiplier: value.image.size.width / value.image.size.height
+                )
+                aspectRatioConstraint?.isActive = true
+            }
             onSuccess?([
                 "width": value.image.size.width,
                 "height": value.image.size.height,
                 "source": value.urlResponse?.url?.absoluteString ?? ""
             ])
+
         case .failure(let error):
             onError?([
                 "error": error.localizedDescription,
diff --git a/node_modules/@candlefinance/faster-image/lib/typescript/index.d.ts b/node_modules/@candlefinance/faster-image/lib/typescript/index.d.ts
index 03be2a8..5b191c6 100644
--- a/node_modules/@candlefinance/faster-image/lib/typescript/index.d.ts
+++ b/node_modules/@candlefinance/faster-image/lib/typescript/index.d.ts
@@ -1,5 +1,5 @@
 import { ImageStyle, ColorValue } from 'react-native';
-export type IOSImageResizeMode = 'fill' | 'contain' | 'cover' | 'center' | 'top' | 'bottom' | 'left' | 'right' | 'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight';
+export type IOSImageResizeMode = 'fill' | 'contain' | 'cover' | 'center' | 'top' | 'topContain' | 'bottom' | 'left' | 'right' | 'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight';
 export type AndroidImageResizeMode = 'fill' | 'contain' | 'cover' | 'center' | 'top' | 'bottom';
 export type ImageOptions = {
     blurhash?: string;
diff --git a/node_modules/@candlefinance/faster-image/src/index.tsx b/node_modules/@candlefinance/faster-image/src/index.tsx
index bb203a8..a0706f3 100644
--- a/node_modules/@candlefinance/faster-image/src/index.tsx
+++ b/node_modules/@candlefinance/faster-image/src/index.tsx
@@ -15,6 +15,7 @@ export type IOSImageResizeMode =
   | 'bottom'
   | 'left'
   | 'right'
+  | 'topContain'
   | 'topLeft'
   | 'topRight'
   | 'bottomLeft'
