diff --git a/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift b/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
index a0b0ac6..e093754 100644
--- a/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
+++ b/node_modules/@candlefinance/faster-image/ios/FasterImageViewManager.swift
@@ -41,7 +41,10 @@ final class FasterImageView: UIView {
             lazyImageView.trailingAnchor.constraint(equalTo: trailingAnchor),
         ])
         lazyImageView.pipeline = .shared
-        lazyImageView.priority = .high
+        // 🌈🌈
+        lazyImageView.priority = .veryLow
+//        lazyImageView.priority = .high
+        // 🌈🌈
         lazyImageView.onCompletion = { [weak self] result in
             self?.completionHandler(with: result)
         }
@@ -121,11 +124,7 @@ final class FasterImageView: UIView {
         }
     }
 
-    var showActivityIndicator = false {
-        didSet {
-            lazyImageView.placeholderView = UIActivityIndicatorView()
-        }
-    }
+    var showActivityIndicator = false
 
     var resizeMode = "contain" {
         didSet {
