diff --git a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetScrollable/createBottomSheetScrollableComponent.tsx b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetScrollable/createBottomSheetScrollableComponent.tsx
index c6a7507..1d2468f 100644
--- a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetScrollable/createBottomSheetScrollableComponent.tsx
+++ b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetScrollable/createBottomSheetScrollableComponent.tsx
@@ -47,7 +47,7 @@ export function createBottomSheetScrollableComponent<T, P>(
 
     //#region hooks
     const { scrollableRef, scrollableContentOffsetY, scrollHandler } =
-      useScrollHandler(scrollEventsHandlersHook);
+      useScrollHandler(scrollEventsHandlersHook, rest.onScrollWorklet);
     const {
       enableContentPanningGesture,
       animatedFooterHeight,
diff --git a/node_modules/@gorhom/bottom-sheet/src/hooks/useScrollHandler.ts b/node_modules/@gorhom/bottom-sheet/src/hooks/useScrollHandler.ts
index 8acc126..262f682 100644
--- a/node_modules/@gorhom/bottom-sheet/src/hooks/useScrollHandler.ts
+++ b/node_modules/@gorhom/bottom-sheet/src/hooks/useScrollHandler.ts
@@ -8,7 +8,8 @@ import { workletNoop as noop } from '../utilities';
 import type { Scrollable } from '../types';
 
 export const useScrollHandler = (
-  useScrollEventsHandlers = useScrollEventsHandlersDefault
+  useScrollEventsHandlers = useScrollEventsHandlersDefault,
+  onScroll
 ) => {
   // refs
   const scrollableRef = useAnimatedRef<Scrollable>();
@@ -28,7 +29,12 @@ export const useScrollHandler = (
   // callbacks
   const scrollHandler = useAnimatedScrollHandler(
     {
-      onScroll: handleOnScroll,
+      onScroll: (...args) => {
+          if (typeof onScroll === 'function') {
+              onScroll(...args)
+          }
+        handleOnScroll(...args)
+      },
       onBeginDrag: handleOnBeginDrag,
       onEndDrag: handleOnEndDrag,
       onMomentumBegin: handleOnMomentumBegin,
