diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
index 1144799..9489fde 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
@@ -86,8 +86,8 @@ exports.setReconnectionConfig = void 0;
 var hw_transport_1 = __importDefault(require("@ledgerhq/hw-transport"));
 var react_native_ble_plx_1 = require("react-native-ble-plx");
 var devices_1 = require("@ledgerhq/devices");
-var sendAPDU_1 = require("@ledgerhq/devices/ble/sendAPDU");
-var receiveAPDU_1 = require("@ledgerhq/devices/ble/receiveAPDU");
+var sendAPDU_1 = require("@ledgerhq/devices/lib/ble/sendAPDU");
+var receiveAPDU_1 = require("@ledgerhq/devices/lib/ble/receiveAPDU");
 var logs_1 = require("@ledgerhq/logs");
 var rxjs_1 = require("rxjs");
 var operators_1 = require("rxjs/operators");