diff --git a/node_modules/@rainbow-me/react-native-payments/lib/ios/ReactNativePayments.m b/node_modules/@rainbow-me/react-native-payments/lib/ios/ReactNativePayments.m
index 2c8395f..57ee90f 100644
--- a/node_modules/@rainbow-me/react-native-payments/lib/ios/ReactNativePayments.m
+++ b/node_modules/@rainbow-me/react-native-payments/lib/ios/ReactNativePayments.m
@@ -2,7 +2,9 @@
 #import <React/RCTUtils.h>
 #import <React/RCTEventDispatcher.h>
 
-@implementation ReactNativePayments
+@implementation ReactNativePayments {
+  BOOL _presented;
+}
 @synthesize bridge = _bridge;
 
 - (dispatch_queue_t)methodQueue
@@ -10,6 +12,15 @@ - (dispatch_queue_t)methodQueue
     return dispatch_get_main_queue();
 }
 
+- (instancetype)init
+{
+  self = [super init];
+  if (self) {
+    _presented = NO;
+  }
+  return self;
+}
+
 RCT_EXPORT_MODULE()
 
 + (BOOL)requiresMainQueueSetup
@@ -66,6 +77,10 @@ - (NSDictionary *)constantsToExport
 
 RCT_EXPORT_METHOD(show:(RCTResponseSenderBlock)callback)
 {
+    if (_presented) {
+      return;
+    }
+    _presented = YES;
     
     self.viewController = [[PKPaymentAuthorizationViewController alloc] initWithPaymentRequest: self.paymentRequest];
     self.viewController.delegate = self;
@@ -74,6 +89,7 @@ - (NSDictionary *)constantsToExport
         UIViewController *rootViewController = RCTPresentedViewController();
         [rootViewController presentViewController:self.viewController animated:YES completion:nil];
         callback(@[[NSNull null]]);
+        self->_presented = NO;
     });
 }
 
