diff --git a/node_modules/@rainbow-me/react-native-payments/android/build.gradle b/node_modules/@rainbow-me/react-native-payments/android/build.gradle
index e2229cd..0d066bd 100644
--- a/node_modules/@rainbow-me/react-native-payments/android/build.gradle
+++ b/node_modules/@rainbow-me/react-native-payments/android/build.gradle
@@ -5,8 +5,8 @@ android {
     buildToolsVersion "28.0.3"
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 26
+        minSdkVersion 21
+        targetSdkVersion 28
         versionCode 1
         versionName "1.0"
         ndk {
@@ -19,6 +19,9 @@ android {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
-    compile 'com.android.support:support-v4:28.0.3'
+    implementation 'com.facebook.react:react-native:+'
+    implementation 'com.google.android.gms:play-services-base:17.0.0'
+    implementation 'com.google.android.gms:play-services-identity:17.0.0'
+    implementation 'com.google.android.gms:play-services-wallet:17.0.0'
+    implementation 'com.android.support:support-v4:23.0.1'
 }
