diff --git a/node_modules/@react-native-menu/menu/.DS_Store b/node_modules/@react-native-menu/menu/.DS_Store
new file mode 100644
index 0000000..c68f0ac
Binary files /dev/null and b/node_modules/@react-native-menu/menu/.DS_Store differ
diff --git a/node_modules/@react-native-menu/menu/android/.DS_Store b/node_modules/@react-native-menu/menu/android/.DS_Store
new file mode 100644
index 0000000..b67a68d
Binary files /dev/null and b/node_modules/@react-native-menu/menu/android/.DS_Store differ
diff --git a/node_modules/@react-native-menu/menu/android/build.gradle b/node_modules/@react-native-menu/menu/android/build.gradle
index dbd6a14..3caad9f 100644
--- a/node_modules/@react-native-menu/menu/android/build.gradle
+++ b/node_modules/@react-native-menu/menu/android/build.gradle
@@ -17,6 +17,8 @@ buildscript {
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
+def DEFAULT_SUPPORT_LIB_VERSION             = "26.1.0"
+
 def getExtOrDefault(name) {
   return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['Menu_' + name]
 }
diff --git a/node_modules/@react-native-menu/menu/android/src/.DS_Store b/node_modules/@react-native-menu/menu/android/src/.DS_Store
new file mode 100644
index 0000000..1c040d0
Binary files /dev/null and b/node_modules/@react-native-menu/menu/android/src/.DS_Store differ
diff --git a/node_modules/@react-native-menu/menu/android/src/main/.DS_Store b/node_modules/@react-native-menu/menu/android/src/main/.DS_Store
new file mode 100644
index 0000000..5480bfe
Binary files /dev/null and b/node_modules/@react-native-menu/menu/android/src/main/.DS_Store differ
diff --git a/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt b/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
index 1cec798..ebdd3a3 100644
--- a/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
+++ b/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
@@ -92,6 +92,12 @@ class MenuView(private val mContext: ReactContext): ReactViewGroup(mContext) {
       true -> config.getMap("attributes")
       else -> null
     }
+
+    val disabled = when (config != null && config.hasKey("disabled") && !config.isNull(("disabled"))) {
+      true -> config.getBoolean("disabled")
+      else -> false
+    }
+
     val subactions = when (config != null && config.hasKey("subactions") && !config.isNull(("subactions"))) {
       true -> config.getArray("subactions")
       else -> null
@@ -101,6 +107,8 @@ class MenuView(private val mContext: ReactContext): ReactViewGroup(mContext) {
       menuItem.title = getMenuItemTextWithColor(menuItem.title.toString(), titleColor)
     }
 
+    menuItem.setEnabled(!disabled)
+
     if (imageName != null) {
       val resourceId: Int = getDrawableIdWithName(imageName)
       if (resourceId != 0) {
