diff --git a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
index 913218e..0276c1b 100755
--- a/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
+++ b/node_modules/@react-navigation/stack/src/views/Stack/Card.tsx
@@ -336,6 +336,11 @@ export default class Card extends React.Component<Props> {
         onGestureEnd?.();
         break;
       }
+      case GestureState.ACTIVE: {
+        // @ts-ignore
+        global.__rainbowDismissScreen()
+        break;
+      }
     }
   };
 
