diff --git a/node_modules/@react-navigation/routers/src/StackRouter.tsx b/node_modules/@react-navigation/routers/src/StackRouter.tsx
index eaa43a0..7efff5e 100644
--- a/node_modules/@react-navigation/routers/src/StackRouter.tsx
+++ b/node_modules/@react-navigation/routers/src/StackRouter.tsx
@@ -483,6 +483,33 @@ export default function StackRouter(options: StackRouterOptions) {
 
           return null;
 
+        // @ts-ignore
+        case '@RAINBOW/REMOVE_NEXT_TO_LAST': {
+          if (state.routes.length < 3) {
+            return state;
+          }
+
+          return {
+            ...state,
+            index: state.index - 1,
+            routes: [
+              ...state.routes.slice(0, -2),
+              state.routes[state.routes.length - 1],
+            ],
+          };
+        }
+
+        // @ts-ignore
+        case '@RAINBOW/REMOVE_FIRST': {
+          return {
+            ...state,
+            index: state.index - 1,
+            routes: [
+              ...state.routes.slice(1),
+            ],
+          };
+        }
+
         default:
           return BaseRouter.getStateForAction(state, action);
       }
