diff --git a/node_modules/@shopify/react-native-performance/android/src/main/kotlin/com/shopify/reactnativeperformance/ReactNativePerformanceModule.kt b/node_modules/@shopify/react-native-performance/android/src/main/kotlin/com/shopify/reactnativeperformance/ReactNativePerformanceModule.kt
index 13fa3f6..6f9879c 100644
--- a/node_modules/@shopify/react-native-performance/android/src/main/kotlin/com/shopify/reactnativeperformance/ReactNativePerformanceModule.kt
+++ b/node_modules/@shopify/react-native-performance/android/src/main/kotlin/com/shopify/reactnativeperformance/ReactNativePerformanceModule.kt
@@ -39,4 +39,7 @@ class ReactNativePerformanceModule(reactContext: ReactContext) :
     fun getNativeUuid(promise: Promise) {
         promise.resolve(UUID.randomUUID().toString())
     }
+
+    override fun getConstants(): MutableMap<String, Any> =
+        hashMapOf("startupTimestamp" to ReactNativePerformance.nativeStartupTimestamp.toString())
 }
diff --git a/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformanceModule.swift b/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformanceModule.swift
index db07d54..a0f43e2 100644
--- a/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformanceModule.swift
+++ b/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformanceModule.swift
@@ -26,6 +26,10 @@ class ReactNativePerformanceModule: NSObject {
         resolve(NSUUID().uuidString)
     }
 
+    @objc func constantsToExport() -> [String: Any] {
+        return ["startupTimestamp": String(ReactNativePerformance.startupTimestamp())]
+    }
+
     @objc static func requiresMainQueueSetup() -> Bool {
         return true
     }
