diff --git a/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/MetalWindowContext.mm b/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/MetalWindowContext.mm
index 3829faf..5be93ba 100644
--- a/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/MetalWindowContext.mm
+++ b/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/MetalWindowContext.mm
@@ -19,6 +19,15 @@
   _layer.pixelFormat = MTLPixelFormatBGRA8Unorm;
   _layer.contentsGravity = kCAGravityBottomLeft;
   _layer.drawableSize = CGSizeMake(width, height);
+  BOOL supportsWideColor = NO;
+  if (@available(iOS 10.0, *)) {
+    supportsWideColor = [UIScreen mainScreen].traitCollection.displayGamut == UIDisplayGamutP3;
+  }
+  if (supportsWideColor) {
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3);
+    _layer.colorspace = colorSpace;
+    CGColorSpaceRelease(colorSpace);
+  }
 }
 
 sk_sp<SkSurface> MetalWindowContext::getSurface() {
