diff --git a/node_modules/@walletconnect/socket-transport/dist/cjs/index.js b/node_modules/@walletconnect/socket-transport/dist/cjs/index.js
index a4fa878..f604389 100644
--- a/node_modules/@walletconnect/socket-transport/dist/cjs/index.js
+++ b/node_modules/@walletconnect/socket-transport/dist/cjs/index.js
@@ -90,6 +90,7 @@ class SocketTransport {
         }
         this._nextSocket.onmessage = (event) => this._socketReceive(event);
         this._nextSocket.onopen = () => this._socketOpen();
+        this._nextSocket.onerror = (e) => this._socketCreate();
     }
     _socketOpen() {
         this._socketClose();
