diff --git a/node_modules/lottie-react-native/lottie-react-native.podspec b/node_modules/lottie-react-native/lottie-react-native.podspec
index 006d118..63caf4a 100644
--- a/node_modules/lottie-react-native/lottie-react-native.podspec
+++ b/node_modules/lottie-react-native/lottie-react-native.podspec
@@ -14,7 +14,11 @@ Pod::Spec.new do |s|
 
   s.source       = { :git => "https://github.com/react-community/lottie-react-native.git", :tag => "v#{s.version}" }
   s.source_files  = "src/ios/**/*.{h,m,swift}"
-  s.swift_version = "5.0"
-  s.dependency 'React'
+  
+  s.requires_arc = true
+  
+  s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES' }
+  
+  s.dependency 'React-Core'
   s.dependency 'lottie-ios', '~> 3.1.8'
 end
diff --git a/node_modules/lottie-react-native/src/ios/LottieReactNative/AnimationViewManagerModule.swift b/node_modules/lottie-react-native/src/ios/LottieReactNative/AnimationViewManagerModule.swift
index 77dbfe2..696f220 100644
--- a/node_modules/lottie-react-native/src/ios/LottieReactNative/AnimationViewManagerModule.swift
+++ b/node_modules/lottie-react-native/src/ios/LottieReactNative/AnimationViewManagerModule.swift
@@ -1,7 +1,3 @@
-#if canImport(React)
-import React
-#endif
-
 import Lottie
 
 @objc(LottieAnimationView)
diff --git a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
index 6d7537b..db15cc5 100644
--- a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
@@ -1,7 +1,3 @@
-#if canImport(React)
-import React
-#endif
-
 import Lottie
 
 class ContainerView: RCTView {
