diff --git a/node_modules/react-native-keyboard-area/android/build.gradle b/node_modules/react-native-keyboard-area/android/build.gradle
deleted file mode 100644
index d4a2cb8..0000000
--- a/node_modules/react-native-keyboard-area/android/build.gradle
+++ /dev/null
@@ -1,42 +0,0 @@
-buildscript {
-    ext {
-        kotlinVersion = '1.3.11'
-        androidToolsVersion = '3.2.1'
-    }
-
-    repositories {
-        google()
-        mavenCentral()
-    }
-
-    dependencies {
-        classpath "com.android.tools.build:gradle:${project.ext.androidToolsVersion}"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${project.ext.kotlinVersion}"
-    }
-}
-
-apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
-
-repositories {
-    google()
-}
-
-android {
-    compileSdkVersion 28
-
-    defaultConfig {
-        minSdkVersion 19
-        targetSdkVersion 28
-        versionCode 1
-        versionName "1.0.0"
-    }
-    lintOptions {
-        abortOnError false
-    }
-}
-
-dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${project.ext.kotlinVersion}"
-    implementation "com.facebook.react:react-native:+"  // From node_modules
-}
\ No newline at end of file
diff --git a/node_modules/react-native-keyboard-area/android/src/main/AndroidManifest.xml b/node_modules/react-native-keyboard-area/android/src/main/AndroidManifest.xml
deleted file mode 100644
index 8de0eeb..0000000
--- a/node_modules/react-native-keyboard-area/android/src/main/AndroidManifest.xml
+++ /dev/null
@@ -1,5 +0,0 @@
-
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-                package="com.rn.keyboard">
-
-</manifest>
diff --git a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/KeyboardProvider.kt b/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/KeyboardProvider.kt
deleted file mode 100644
index 37103f0..0000000
--- a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/KeyboardProvider.kt
+++ /dev/null
@@ -1,96 +0,0 @@
-package com.rn.keyboard
-
-import android.app.Activity
-import android.graphics.Rect
-import android.view.Gravity
-import android.view.View
-import android.view.ViewTreeObserver
-import android.view.WindowManager
-import android.widget.PopupWindow
-
-object KeyboardInfo {
-    const val HEIGHT_NOT_COMPUTED = -1
-    const val STATE_UNKNOWN = -1
-    const val STATE_CLOSED = 0
-    const val STATE_OPENED = 1
-
-    /**
-     * Cached keyboard height. This will keep the last keyboard height value and not
-     * it's current height
-     */
-    var keyboardHeight = HEIGHT_NOT_COMPUTED
-
-    /**
-     * Real time keyboard state
-     */
-    var keyboardState = STATE_UNKNOWN
-}
-
-class KeyboardProvider(private val activity: Activity) : PopupWindow(activity) {
-
-    private var heightMax = 0;
-    private var lastKeyboardHeight = -1
-    private var resizableView: View
-    private var parentView: View? = null
-    private var keyboardListeners = ArrayList<KeyboardListener>()
-
-    init {
-        contentView = View.inflate(activity, R.layout.keyboard_popup, null)
-        resizableView = contentView.findViewById(R.id.keyResizeContainer)
-        softInputMode = WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE or
-                WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE
-        inputMethodMode = PopupWindow.INPUT_METHOD_NEEDED
-
-        width = 0
-        height = WindowManager.LayoutParams.MATCH_PARENT
-
-        parentView = activity.window.decorView.rootView
-        parentView?.post {
-            resizableView.viewTreeObserver.addOnGlobalLayoutListener(getGlobalLayoutListener())
-            if (!isShowing && parentView?.windowToken != null) {
-                showAtLocation(parentView, Gravity.NO_GRAVITY, 0, 0)
-            }
-        }
-    }
-
-    fun addKeyboardListener(listener: KeyboardListener) {
-        keyboardListeners.add(listener)
-    }
-
-    fun removeKeyboardListener() {
-        keyboardListeners.clear()
-    }
-
-    private fun getGlobalLayoutListener() = ViewTreeObserver.OnGlobalLayoutListener {
-        computeKeyboardState()
-    }
-
-    private fun computeKeyboardState() {
-        val rect = Rect()
-        contentView.rootView.getWindowVisibleDisplayFrame(rect)
-        val orientation = activity.resources.configuration.orientation
-        
-        if (rect.bottom > heightMax) {
-            heightMax = rect.bottom;
-        }
-        val keyboardHeight = heightMax - rect.bottom
-        KeyboardInfo.keyboardState = if (keyboardHeight > 0) KeyboardInfo.STATE_OPENED else KeyboardInfo.STATE_CLOSED
-        if (keyboardHeight > 0) {
-            KeyboardInfo.keyboardHeight = keyboardHeight
-        }
-        if (keyboardHeight != lastKeyboardHeight) {
-            notifyKeyboardHeightChanged(keyboardHeight, orientation)
-        }
-        lastKeyboardHeight = keyboardHeight
-    }
-
-    private fun notifyKeyboardHeightChanged(height: Int, orientation: Int) {
-        keyboardListeners.forEach {
-            it.onHeightChanged(height)
-        }
-    }
-
-    interface KeyboardListener {
-        fun onHeightChanged(height: Int)
-    }
-}
\ No newline at end of file
diff --git a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardModule.kt b/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardModule.kt
deleted file mode 100644
index 7217fdb..0000000
--- a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardModule.kt
+++ /dev/null
@@ -1,85 +0,0 @@
-package com.rn.keyboard
-
-import android.util.Log
-import android.view.WindowManager.LayoutParams
-import com.facebook.react.bridge.*
-import com.facebook.react.modules.core.DeviceEventManagerModule
-
-@Suppress("unused")
-class RNKeyboardModule(private val reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
-    private var keyboardProvider: KeyboardProvider? = null
-
-    @ReactMethod
-    fun setWindowSoftInputMode(mode: Int, promise: Promise) {
-        try {
-            UiThreadUtil.runOnUiThread {
-                currentActivity?.window?.setSoftInputMode(mode)
-                promise.resolve("success")
-            }
-        } catch (e: Exception) {
-            promise.reject("error", e.toString())
-        }
-    }
-
-    @ReactMethod
-    fun startKeyboardListener() {
-        try {
-            UiThreadUtil.runOnUiThread {
-                val mActivity = currentActivity
-                if(mActivity != null) {
-                    keyboardProvider = KeyboardProvider(mActivity)
-                    keyboardProvider?.addKeyboardListener(object : KeyboardProvider.KeyboardListener{
-                        override fun onHeightChanged(height: Int) {
-                            emit(height)
-                        }
-                    })
-                }
-            }
-        } catch (e: Exception) {
-            e.printStackTrace()
-        }
-    }
-
-    @ReactMethod
-    fun stopKeyboardListener() {
-        try {
-            keyboardProvider?.removeKeyboardListener();
-        } catch (e: Exception) {
-            e.printStackTrace()
-        }
-    }
-
-    override fun getName(): String {
-        return "RNKeyboard"
-    }
-
-    fun emit(height: Int) {
-        try {
-            reactApplicationContext
-                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-                    .emit("KeyboardSizeChanges", height)
-        } catch (e: InterruptedException) {
-            e.printStackTrace()
-        }
-    }
-
-    override fun getConstants(): Map<String, Any>? {
-        var constants = HashMap<String, Any>()
-        constants.put("SOFT_INPUT_ADJUST_NOTHING", LayoutParams.SOFT_INPUT_ADJUST_NOTHING)
-        constants.put("SOFT_INPUT_ADJUST_PAN", LayoutParams.SOFT_INPUT_ADJUST_PAN)
-        constants.put("SOFT_INPUT_ADJUST_RESIZE", LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
-        constants.put("SOFT_INPUT_ADJUST_UNSPECIFIED", LayoutParams.SOFT_INPUT_ADJUST_UNSPECIFIED)
-        constants.put("SOFT_INPUT_IS_FORWARD_NAVIGATION", LayoutParams.SOFT_INPUT_IS_FORWARD_NAVIGATION)
-        constants.put("SOFT_INPUT_MASK_ADJUST", LayoutParams.SOFT_INPUT_MASK_ADJUST)
-        constants.put("SOFT_INPUT_MASK_STATE", LayoutParams.SOFT_INPUT_MASK_STATE)
-        constants.put("SOFT_INPUT_MODE_CHANGED", LayoutParams.SOFT_INPUT_MODE_CHANGED)
-        constants.put("SOFT_INPUT_STATE_ALWAYS_HIDDEN", LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)
-        constants.put("SOFT_INPUT_STATE_ALWAYS_VISIBLE", LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE)
-        constants.put("SOFT_INPUT_STATE_HIDDEN", LayoutParams.SOFT_INPUT_STATE_HIDDEN)
-        constants.put("SOFT_INPUT_STATE_UNCHANGED", LayoutParams.SOFT_INPUT_STATE_UNCHANGED)
-        constants.put("SOFT_INPUT_STATE_UNSPECIFIED", LayoutParams.SOFT_INPUT_STATE_UNSPECIFIED)
-        constants.put("SOFT_INPUT_STATE_VISIBLE", LayoutParams.SOFT_INPUT_STATE_VISIBLE)
-        return constants;
-    }
-
-}
diff --git a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardPackage.kt b/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardPackage.kt
deleted file mode 100644
index 2b91181..0000000
--- a/node_modules/react-native-keyboard-area/android/src/main/java/com/rn/keyboard/RNKeyboardPackage.kt
+++ /dev/null
@@ -1,21 +0,0 @@
-package com.rn.keyboard
-
-import android.view.View
-import com.facebook.react.ReactPackage
-import com.facebook.react.bridge.NativeModule
-import com.facebook.react.bridge.ReactApplicationContext
-import com.facebook.react.uimanager.ViewManager
-import com.facebook.react.uimanager.ReactShadowNode
-
-class RNKeyboardPackage : ReactPackage {
-
-    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
-        return listOf(
-            RNKeyboardModule(reactContext)
-        )
-    }
-
-    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<out View, out ReactShadowNode<*>>> {
-        return listOf()
-    }
-}
\ No newline at end of file
diff --git a/node_modules/react-native-keyboard-area/android/src/main/res/layout/keyboard_popup.xml b/node_modules/react-native-keyboard-area/android/src/main/res/layout/keyboard_popup.xml
deleted file mode 100644
index d0d172d..0000000
--- a/node_modules/react-native-keyboard-area/android/src/main/res/layout/keyboard_popup.xml
+++ /dev/null
@@ -1,7 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/keyResizeContainer"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent">
-
-</FrameLayout>
\ No newline at end of file
