diff --git a/node_modules/react-native-svg/.DS_Store b/node_modules/react-native-svg/.DS_Store
new file mode 100644
index 0000000..79740aa
Binary files /dev/null and b/node_modules/react-native-svg/.DS_Store differ
diff --git a/node_modules/react-native-svg/ios/Elements/RNSVGGroup.m b/node_modules/react-native-svg/ios/Elements/RNSVGGroup.m
index a8787b8..22f6259 100644
--- a/node_modules/react-native-svg/ios/Elements/RNSVGGroup.m
+++ b/node_modules/react-native-svg/ios/Elements/RNSVGGroup.m
@@ -148,7 +148,6 @@ - (CGPathRef)getPath:(CGContextRef)context
     [self traverseSubviews:^(RNSVGNode *node) {
         if ([node isKindOfClass:[RNSVGNode class]] && ![node isKindOfClass:[RNSVGMask class]]) {
             CGAffineTransform transform = CGAffineTransformConcat(node.matrix, node.transforms);
-            CGPathAddPath(path, &transform, [node getPath:context]);
             CGPathAddPath(path, &transform, [node markerPath]);
             node.dirty = false;
         }
diff --git a/node_modules/react-native-svg/ios/RNSVGRenderable.m b/node_modules/react-native-svg/ios/RNSVGRenderable.m
index 5fddadd..635f9b1 100644
--- a/node_modules/react-native-svg/ios/RNSVGRenderable.m
+++ b/node_modules/react-native-svg/ios/RNSVGRenderable.m
@@ -374,6 +374,7 @@ - (void)renderLayerTo:(CGContextRef)context rect:(CGRect)rect
     if (_vectorEffect == kRNSVGVectorEffectNonScalingStroke) {
         path = CGPathCreateCopyByTransformingPath(path, &svgToClientTransform);
         CGContextConcatCTM(context, CGAffineTransformInvert(svgToClientTransform));
+        CFAutorelease(path);
     }
 
     CGAffineTransform vbmatrix = self.svgView.getViewBoxTransform;
@@ -486,6 +487,7 @@ - (void)renderLayerTo:(CGContextRef)context rect:(CGRect)rect
 
     CGContextAddPath(context, path);
     CGContextDrawPath(context, mode);
+
 }
 
 - (void)setHitArea:(CGPathRef)path
