diff --git a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/DataStorePrefsStorage.kt b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/DataStorePrefsStorage.kt
index 70912a9..006f7d8 100644
--- a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/DataStorePrefsStorage.kt
+++ b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/DataStorePrefsStorage.kt
@@ -48,6 +48,32 @@ class DataStorePrefsStorage(
     return ResultSet(cipherStorageName, bytesForUsername, bytesForPassword)
   }
 
+  override fun getAllEncryptedEntries(): MutableList<Map<String, Any?>> {
+    val processedServices = mutableListOf<String>()
+    val data = mutableListOf<Map<String, Any?>>()
+    runBlocking {
+      val allKeys = prefsData.asMap().keys
+      for (key in allKeys) {
+        val keyName = key.name
+        if (keyName.length >= 2) {
+          val serviceName = keyName.substring(0, keyName.length - 2)
+          if (!processedServices.contains(serviceName)) {
+            val item = getEncryptedEntry(serviceName)
+            if (item != null) {
+              val map = mapOf(
+                "resultSet" to item,
+                "service" to serviceName
+              )
+              data.add(map)
+              processedServices.add(serviceName)
+            }
+          }
+        }
+      }
+    }
+    return data
+  }
+
   override fun removeEntry(service: String) {
     val keyForUsername = stringPreferencesKey(getKeyForUsername(service))
     val keyForPassword = stringPreferencesKey(getKeyForPassword(service))
diff --git a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/KeychainModule.kt b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/KeychainModule.kt
index 722a5ed..311a2f4 100644
--- a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/KeychainModule.kt
+++ b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/KeychainModule.kt
@@ -281,6 +281,49 @@ class KeychainModule(reactContext: ReactApplicationContext) :
     }
   }
 
+  @ReactMethod
+  fun getAllInternetCredentialsForServer(options: ReadableMap, promise: Promise) {
+    coroutineScope.launch {
+      mutex.withLock {
+        try {
+          val accessControl = getAccessControlOrDefault(options)
+          val usePasscode = getUsePasscode(accessControl) && isPasscodeAvailable
+          val useBiometry =
+            getUseBiometry(accessControl) && (isFingerprintAuthAvailable || isFaceAuthAvailable || isIrisAuthAvailable)
+          val allCredentials = Arguments.createArray()
+          val currentCipherStorage = getCipherStorageForCurrentAPILevel(useBiometry, usePasscode)
+          val allResults = prefsStorage.getAllEncryptedEntries()
+
+          for (data in allResults) {
+            val resultSet = data["resultSet"] as PrefsStorageBase.ResultSet
+            val service = data["service"] as String
+            val promptInfo = getPromptInfo(options, usePasscode, useBiometry)
+            val decryptionResult =
+              decryptCredentials(service, currentCipherStorage, resultSet, promptInfo)
+
+            val credentials = Arguments.createMap().apply {
+              putString("service", service)
+              putString("username", decryptionResult.username)
+              putString("password", decryptionResult.password)
+            }
+            allCredentials.pushMap(credentials)
+          }
+
+          val result = Arguments.createMap().apply {
+            putArray("results", allCredentials)
+          }
+          promise.resolve(result)
+        } catch (e: KeyStoreAccessException) {
+          e.message?.let { Log.e(KEYCHAIN_MODULE, it) }
+          promise.reject(Errors.E_KEYSTORE_ACCESS_ERROR, e)
+        } catch (e: CryptoFailedException) {
+          e.message?.let { Log.e(KEYCHAIN_MODULE, it) }
+          promise.reject(Errors.E_CRYPTO_FAILED, e)
+        }
+      }
+    }
+  }
+
   @ReactMethod
   fun getAllGenericPasswordServices(options: ReadableMap?, promise: Promise) {
     try {
@@ -767,19 +810,13 @@ class KeychainModule(reactContext: ReactApplicationContext) :
           BiometricManager.Authenticators.DEVICE_CREDENTIAL
 
         else ->
-          null
+          BiometricManager.Authenticators.BIOMETRIC_STRONG or BiometricManager.Authenticators.DEVICE_CREDENTIAL
       }
 
       if (allowedAuthenticators != null) {
         promptInfoBuilder.setAllowedAuthenticators(allowedAuthenticators)
       }
 
-      if (!usePasscode) {
-        promptInfoOptionsMap?.getString(AuthPromptOptions.CANCEL)?.let {
-          promptInfoBuilder.setNegativeButtonText(it)
-        }
-      }
-
       /* Bypass confirmation to avoid KeyStore unlock timeout being exceeded when using passive biometrics */
       promptInfoBuilder.setConfirmationRequired(false)
       return promptInfoBuilder.build()
diff --git a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/PrefsStorageBase.kt b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/PrefsStorageBase.kt
index d165f3e..91fd8b1 100644
--- a/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/PrefsStorageBase.kt
+++ b/node_modules/react-native-keychain/android/src/main/java/com/oblador/keychain/PrefsStorageBase.kt
@@ -13,6 +13,8 @@ interface PrefsStorageBase {
 
   fun getEncryptedEntry(service: String): ResultSet?
 
+  fun getAllEncryptedEntries(): MutableList<Map<String, Any?>>
+
   fun removeEntry(service: String)
 
   fun storeEncryptedEntry(service: String, encryptionResult: EncryptionResult)
diff --git a/node_modules/react-native-keychain/ios/RNKeychainManager/RNKeychainManager.m b/node_modules/react-native-keychain/ios/RNKeychainManager/RNKeychainManager.m
index 6888ff0..23675f3 100644
--- a/node_modules/react-native-keychain/ios/RNKeychainManager/RNKeychainManager.m
+++ b/node_modules/react-native-keychain/ios/RNKeychainManager/RNKeychainManager.m
@@ -519,6 +519,86 @@ - (OSStatus)deleteCredentialsForServer:(NSString *)server withOptions:(NSDiction
   return resolve([result copy]);
 }
 
+RCT_EXPORT_METHOD(getAllInternetCredentialsForServer:(NSDictionary *)options resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+  NSDictionary *query = @{
+    (__bridge NSString *)kSecClass: (__bridge id)(kSecClassInternetPassword),
+    (__bridge NSString *)kSecReturnAttributes: (__bridge id)kCFBooleanTrue,
+    (__bridge NSString *)kSecReturnData: (__bridge id)kCFBooleanTrue,
+    (__bridge NSString *)kSecMatchLimit: (__bridge NSString *)kSecMatchLimitAll
+  };
+
+  // Look up server in the keychain
+  NSArray *found = nil;
+  CFTypeRef foundTypeRef = NULL;
+  OSStatus osStatus = SecItemCopyMatching((__bridge CFDictionaryRef) query, (CFTypeRef*)&foundTypeRef);
+
+  if (osStatus != noErr && osStatus != errSecItemNotFound) {
+    NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:osStatus userInfo:nil];
+    return rejectWithError(reject, error);
+  }
+
+  found = (__bridge NSArray*)(foundTypeRef);
+  if (!found) {
+    return resolve(@(NO));
+  }
+
+  NSMutableArray *results = [@[] mutableCopy];
+
+  for(int i=0; i< found.count; i++){
+      NSDictionary *item  = found[i];
+      NSString *server = (NSString *) [item objectForKey:(__bridge id)(kSecAttrServer)];
+      NSString *username = (NSString *) [item objectForKey:(__bridge id)(kSecAttrAccount)];
+        NSString *password = [[NSString alloc] initWithData:[item objectForKey:(__bridge id)(kSecValueData)] encoding:NSUTF8StringEncoding];
+
+        [results addObject:@{@"username": username, @"password":password, @"server": server}];
+  }
+
+  CFRelease(foundTypeRef);
+
+  return resolve(@{
+    @"results": results
+  });
+}
+
+RCT_EXPORT_METHOD(getAllInternetCredentialsKeys:(NSDictionary *)options resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+  NSDictionary *query = @{
+    (__bridge NSString *)kSecClass: (__bridge id)(kSecClassInternetPassword),
+    (__bridge NSString *)kSecReturnAttributes: (__bridge id)kCFBooleanTrue,
+    (__bridge NSString *)kSecMatchLimit: (__bridge NSString *)kSecMatchLimitAll
+  };
+
+  // Look up all the keys in the keychain
+  NSArray *found = nil;
+  CFTypeRef foundTypeRef = NULL;
+  OSStatus osStatus = SecItemCopyMatching((__bridge CFDictionaryRef) query, (CFTypeRef*)&foundTypeRef);
+
+  if (osStatus != noErr && osStatus != errSecItemNotFound) {
+    NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:osStatus userInfo:nil];
+    return rejectWithError(reject, error);
+  }
+
+  found = (__bridge NSArray*)(foundTypeRef);
+  if (!found) {
+    return resolve(@(NO));
+  }
+
+  NSMutableArray *results = [@[] mutableCopy];
+
+  for(int i=0; i< found.count; i++){
+      NSDictionary *item  = found[i];
+      NSString *username = (NSString *) [item objectForKey:(__bridge id)(kSecAttrAccount)];
+      [results addObject:username];
+  }
+
+  CFRelease(foundTypeRef);
+
+  return resolve(@{
+    @"results": results
+  });
+}
+
 RCT_EXPORT_METHOD(resetGenericPasswordForOptions:(NSDictionary *)options
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
diff --git a/node_modules/react-native-keychain/lib/commonjs/index.js b/node_modules/react-native-keychain/lib/commonjs/index.js
index da7ccb9..9ab42d5 100644
--- a/node_modules/react-native-keychain/lib/commonjs/index.js
+++ b/node_modules/react-native-keychain/lib/commonjs/index.js
@@ -12,6 +12,8 @@ var _exportNames = {
   hasInternetCredentials: true,
   setInternetCredentials: true,
   getInternetCredentials: true,
+  getAllInternetCredentials: true,
+  getAllInternetCredentialsKeys: true,
   resetInternetCredentials: true,
   getSupportedBiometryType: true,
   requestSharedWebCredentials: true,
@@ -25,6 +27,8 @@ exports.default = void 0;
 exports.getAllGenericPasswordServices = getAllGenericPasswordServices;
 exports.getGenericPassword = getGenericPassword;
 exports.getInternetCredentials = getInternetCredentials;
+exports.getAllInternetCredentials = getAllInternetCredentials;
+exports.getAllInternetCredentialsKeys = getAllInternetCredentialsKeys;
 exports.getSecurityLevel = getSecurityLevel;
 exports.getSupportedBiometryType = getSupportedBiometryType;
 exports.hasGenericPassword = hasGenericPassword;
@@ -212,6 +216,24 @@ function getInternetCredentials(server, options) {
   return RNKeychainManager.getInternetCredentialsForServer(server, (0, _normalizeOptions.normalizeAuthPrompt)(options));
 }
 
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[{ server, username, password }, ...]` when successful
+ */
+function getAllInternetCredentials(options) {
+  return RNKeychainManager.getAllInternetCredentialsForServer(normalizeOptions(options));
+}
+
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[username]` when successful
+ */
+function getAllInternetCredentialsKeys(options) {
+  return RNKeychainManager.getAllInternetCredentialsKeys(normalizeOptions(options));
+}
+
 /**
  * Deletes all internet password keychain entries for the given server.
  *
@@ -370,6 +392,8 @@ var _default = exports.default = {
   setInternetCredentials,
   isPasscodeAuthAvailable,
   getInternetCredentials,
+  getAllInternetCredentials,
+  getAllInternetCredentialsKeys,
   resetInternetCredentials,
   setGenericPassword,
   getGenericPassword,
diff --git a/node_modules/react-native-keychain/lib/module/index.js b/node_modules/react-native-keychain/lib/module/index.js
index b4f2bb6..b19530c 100644
--- a/node_modules/react-native-keychain/lib/module/index.js
+++ b/node_modules/react-native-keychain/lib/module/index.js
@@ -153,6 +153,24 @@ export function getInternetCredentials(server, options) {
   return RNKeychainManager.getInternetCredentialsForServer(server, normalizeAuthPrompt(options));
 }
 
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[{ server, username, password }, ...]` when successful
+ */
+export function getAllInternetCredentials(options) {
+  return RNKeychainManager.getAllInternetCredentialsForServer(normalizeOptions(options));
+}
+
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[username]` when successful
+ */
+export function getAllInternetCredentialsKeys(options) {
+  return RNKeychainManager.getAllInternetCredentialsKeys(normalizeOptions(options));
+}
+
 /**
  * Deletes all internet password keychain entries for the given server.
  *
diff --git a/node_modules/react-native-keychain/lib/typescript/index.d.ts b/node_modules/react-native-keychain/lib/typescript/index.d.ts
index b868899..74f547b 100644
--- a/node_modules/react-native-keychain/lib/typescript/index.d.ts
+++ b/node_modules/react-native-keychain/lib/typescript/index.d.ts
@@ -1,5 +1,5 @@
 import { ACCESSIBLE, ACCESS_CONTROL, AUTHENTICATION_TYPE, SECURITY_LEVEL, STORAGE_TYPE, BIOMETRY_TYPE } from './enums';
-import type { Result, UserCredentials, SharedWebCredentials, GetOptions, GetAllOptions, BaseOptions, SetOptions, AuthenticationTypeOption, AccessControlOption } from './types';
+import type { Result, UserCredentials, SharedWebCredentials, GetOptions, GetAllOptions, BaseOptions, SetOptions, AuthenticationTypeOption, AccessControlOption, AllCredentials, AllCredentialsKeys } from './types';
 /**
  * Saves the `username` and `password` combination for the given service.
  *
@@ -135,6 +135,18 @@ export declare function getInternetCredentials(server: string, options?: GetOpti
  * console.log('Credentials reset for server');
  * ```
  */
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[{ server, username, password }, ...]` when successful
+ */
+export declare function getAllInternetCredentials(options?: GetOptions): Promise<false | AllCredentials>;
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[username]` when successful
+ */
+export declare function getAllInternetCredentialsKeys(options?: GetOptions): Promise<false | AllCredentialsKeys>;
 export declare function resetInternetCredentials(options: BaseOptions): Promise<void>;
 /**
  * Gets the type of biometric authentication supported by the device.
diff --git a/node_modules/react-native-keychain/lib/typescript/types.d.ts b/node_modules/react-native-keychain/lib/typescript/types.d.ts
index ff6a12e..e826925 100644
--- a/node_modules/react-native-keychain/lib/typescript/types.d.ts
+++ b/node_modules/react-native-keychain/lib/typescript/types.d.ts
@@ -111,6 +111,21 @@ export type UserCredentials = {
     /** The password associated with the keychain item. */
     password: string;
 } & Result;
+
+/**
+ * All credentials returned by keychain functions.
+ */
+export type AllCredentials = {
+    results: UserCredentials[];
+};
+
+/**
+ * All credentials keys returned by keychain functions.
+ */
+export type AllCredentialsKeys = {
+    results: string[];
+};
+
 /**
  * Shared web credentials returned by keychain functions.
  * @platform iOS
diff --git a/node_modules/react-native-keychain/src/index.ts b/node_modules/react-native-keychain/src/index.ts
index 1de35da..08eeb88 100644
--- a/node_modules/react-native-keychain/src/index.ts
+++ b/node_modules/react-native-keychain/src/index.ts
@@ -17,6 +17,8 @@ import type {
   SetOptions,
   AuthenticationTypeOption,
   AccessControlOption,
+  AllCredentials,
+  AllCredentialsKeys,
 } from './types';
 import { normalizeAuthPrompt } from './normalizeOptions';
 
@@ -200,6 +202,32 @@ export function getInternetCredentials(
   );
 }
 
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[{ server, username, password }, ...]` when successful
+ */
+export function getAllInternetCredentials(
+  options?: GetOptions
+): Promise<false | AllCredentials> {
+  return RNKeychainManager.getAllInternetCredentialsForServer(
+    normalizeAuthPrompt(options)
+  );
+}
+
+/**
+ * Fetches all the keychain entries for the app
+ * @param {object} options Keychain options, iOS only
+ * @return {Promise} Resolves to `[username]` when successful
+ */
+export function getAllInternetCredentialsKeys(
+  options?: GetOptions
+): Promise<false | AllCredentialsKeys> {
+  return RNKeychainManager.getAllInternetCredentialsKeys(
+    normalizeAuthPrompt(options)
+  );
+}
+
 /**
  * Deletes all internet password keychain entries for the given server.
  *
@@ -384,6 +412,8 @@ export default {
   setInternetCredentials,
   isPasscodeAuthAvailable,
   getInternetCredentials,
+  getAllInternetCredentials,
+  getAllInternetCredentialsKeys,
   resetInternetCredentials,
   setGenericPassword,
   getGenericPassword,
diff --git a/node_modules/react-native-keychain/src/normalizeOptions.ts b/node_modules/react-native-keychain/src/normalizeOptions.ts
index e72d2fb..ea4d3c7 100644
--- a/node_modules/react-native-keychain/src/normalizeOptions.ts
+++ b/node_modules/react-native-keychain/src/normalizeOptions.ts
@@ -12,10 +12,11 @@ export function normalizeAuthPrompt<T extends SetOptions | GetOptions>(
 ) {
   const { authenticationPrompt } = options;
 
-  options.authenticationPrompt = {
-    ...AUTH_PROMPT_DEFAULTS,
-    ...authenticationPrompt,
+  return {
+    ...options,
+    authenticationPrompt: {
+      ...AUTH_PROMPT_DEFAULTS,
+      ...authenticationPrompt,
+    },
   };
-
-  return options;
 }
diff --git a/node_modules/react-native-keychain/src/types.ts b/node_modules/react-native-keychain/src/types.ts
index 00448d3..4a6c58f 100644
--- a/node_modules/react-native-keychain/src/types.ts
+++ b/node_modules/react-native-keychain/src/types.ts
@@ -129,6 +129,20 @@ export type UserCredentials = {
   password: string;
 } & Result;
 
+/**
+ * All credentials returned by keychain functions.
+ */
+export type AllCredentials = {
+  results: UserCredentials[];
+};
+
+/**
+ * All credentials keys returned by keychain functions.
+ */
+export type AllCredentialsKeys = {
+  results: string[];
+};
+
 /**
  * Shared web credentials returned by keychain functions.
  * @platform iOS
