--- node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java	2026-05-02 03:41:34
+++ node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSManager.java	2026-05-02 03:41:34
@@ -960,7 +960,7 @@
       return;
     }
 
-    promise.reject(null, ex.getMessage());
+    promise.reject("ERROR", ex.getMessage(), ex);
   }
 
   private void rejectFileNotFound(Promise promise, String filepath) {
