diff --git a/node_modules/react-native-screens/ios/RNSScreenStack.m b/node_modules/react-native-screens/ios/RNSScreenStack.m
index 21172ba..f9e6c29 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStack.m
+++ b/node_modules/react-native-screens/ios/RNSScreenStack.m
@@ -285,6 +285,9 @@ - (void)setModalViewControllers:(NSArray<UIViewController *> *)controllers
       for (NSUInteger i = changeRootIndex; i < controllers.count; i++) {
         UIViewController *next = controllers[i];
         BOOL lastModal = (i == controllers.count - 1);
+        if (previous.presentedViewController) {
+          previous = previous.presentedViewController;
+        }
         [previous presentViewController:next
                                animated:lastModal
                              completion:^{
@@ -298,7 +301,7 @@ - (void)setModalViewControllers:(NSArray<UIViewController *> *)controllers
     }
   };
 
-  if (changeRootController.presentedViewController) {
+  if (changeRootController.presentedViewController && ![NSStringFromClass(changeRootController.presentedViewController.class) isEqualToString:@"reactnativeslackbottomsheet.PanModalViewController"]) {
     [changeRootController
      dismissViewControllerAnimated:(changeRootIndex == controllers.count)
      completion:finish];
