diff --git a/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m b/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
index 637f112..f840ecb 100644
--- a/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
+++ b/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
@@ -39,6 +39,7 @@ - (void)layoutSubviews {
 @end
 
 @interface InvisibleView: RCTView
+- (void)jumpTo:(NSNumber*)longForm;
 @property (nonatomic, nonnull) NSNumber *topOffset;
 @property (nonatomic) BOOL isShortFormEnabled;
 @property (nonatomic, nullable) NSNumber *longFormHeight;
@@ -121,6 +122,12 @@ -(void) setScrollsToTopOnTapStatusBar:(BOOL) scrollsToTop {
   }
 }
 
+-(void)jumpTo:(NSNumber*)longForm {
+  if (_contoller != nil) {
+    [_contoller performSelector:NSSelectorFromString(@"jumpToLong:") withObject:longForm];
+  }
+}
+
 -(void)reactSetFrame:(CGRect)frame {
   // shrug
 }
@@ -182,7 +189,7 @@ - (void)setVisible:(BOOL)visible {
       self->_modalPresented = YES;
     });
   } else {
-    RCTExecuteOnMainQueue(^{
+    //RCTExecuteOnMainQueue(^{
       if (!self->_modalPresented) {
         return;
       }
@@ -193,12 +200,17 @@ - (void)setVisible:(BOOL)visible {
       self.transitionDuration = [[NSNumber alloc] initWithDouble: 0];
       UIViewController *rootViewController = [UIApplication sharedApplication].delegate.window.rootViewController;
 
-      [[rootViewController presentedViewController] dismissViewControllerAnimated:!self->_isHiding completion:^{
+      [[rootViewController presentedViewController] dismissViewControllerAnimated:NO completion:^{
+        if (self.onDidDismiss) {
+          self.onDidDismiss(nil);
+        }
         self.transitionDuration = oldTransitionDuration;
       }];
+      //self->_contoller.view = nil;
+      self->_contoller = nil;
       self->_isHiding = false;
       self->_modalPresented = NO;
-    });
+    //});
   }
 }
 
@@ -213,7 +225,7 @@ - (void)addSubview:(UIView *)view {
 
 @end
 
-@interface ModalViewManager : RCTViewManager
+@interface ModalViewManager : RCTViewManager<RCTBridgeModule>
 @end
 
 @implementation ModalViewManager
@@ -247,6 +259,15 @@ @implementation ModalViewManager
 RCT_EXPORT_VIEW_PROPERTY(unmountAnimation, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(visible, BOOL)
 
+RCT_EXPORT_METHOD(jumpTo:(nonnull NSNumber*)point tag:(nonnull NSNumber*) reactTag) {
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *,UIView *> *viewRegistry) {
+    InvisibleView *view = (InvisibleView *)viewRegistry[reactTag];
+    [view jumpTo:point];
+  }];
+  
+}
+
+
 - (UIView *)view {
   return [[InvisibleView alloc] initWithBridge:self.bridge];
 }
diff --git a/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift b/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
index b688639..2a6f218 100644
--- a/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
+++ b/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
@@ -29,6 +29,8 @@ class BetterGestureRecognizerDelegateAdapter: NSObject, UIGestureRecognizerDeleg
   }
 }
 
+var moved = false
+
 class PossiblyTouchesPassableUIView: UIView {
   var grdelegate: UIGestureRecognizerDelegate?
   var config: NSObject?
@@ -49,12 +51,17 @@ class PossiblyTouchesPassableUIView: UIView {
     super.layoutSubviews()
     let outerView = self.config?.value(forKey: "outerView") as? UIView
     if (!(self.config!.value(forKey: "presentGlobally") as! Bool)) {
+      if moved {
+        return
+      }
+      moved = true
       removeFromSuperview()
       let helperView: UIView = self.subviews[1].subviews[0]
-      let bounds = outerView!.bounds
-      let topOffset:CGFloat = (topLayoutGuideLength ?? 0) + CGFloat(truncating: self.config?.value(forKey: "topOffset") as! NSNumber)
-      let newBounds = CGRect.init(x: bounds.minX, y: bounds.minY, width: bounds.width, height: bounds.height - topOffset)
-      helperView.setValue(newBounds, forKeyPath: "specialBounds")
+        let bounds = outerView!.bounds
+        let topOffset:CGFloat = (topLayoutGuideLength ?? 0) + CGFloat(truncating: self.config?.value(forKey: "topOffset") as! NSNumber)
+
+        let newBounds = CGRect.init(x: bounds.minX, y: bounds.minY, width: bounds.width, height: bounds.height - topOffset)
+        helperView.setValue(newBounds, forKeyPath: "specialBounds")
       outerView?.addSubview(self)
     }
     let gr: UIGestureRecognizer = self.gestureRecognizers![0]
@@ -97,6 +104,16 @@ class PanModalViewController: UIViewController, PanModalPresentable {
     
   }
   
+  @objc func jumpTo(long: NSNumber) {
+    self.panModalSetNeedsLayoutUpdate()
+    if (long.boolValue) {
+      panModalTransition(to: .longForm);
+    } else {
+      panModalTransition(to: .shortForm);
+    }
+    self.panModalSetNeedsLayoutUpdate()
+  }
+  
   var config: NSObject?
   var scrollView: UIScrollView?
   convenience init(config: NSObject) {
@@ -104,6 +121,11 @@ class PanModalViewController: UIViewController, PanModalPresentable {
     self.config = config
   }
 
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    hack()
+  }
+
 
   override var preferredStatusBarStyle: UIStatusBarStyle {
     return .lightContent
@@ -203,6 +225,15 @@ class PanModalViewController: UIViewController, PanModalPresentable {
     return self.config?.value(forKey: "isHapticFeedbackEnabled") as! Bool  }
 
   var shouldRoundTopCorners: Bool {
+    hack()
+    return self.config?.value(forKey: "shouldRoundTopCorners") as! Bool
+  }
+  
+  override func viewDidLayoutSubviews() {
+    hack()
+  }
+  
+  public func hack() {
     let pview = view.superview!.superview!
     if !(pview is PossiblyTouchesPassableUIView) {
       let oldClass: AnyClass = type(of: pview)
@@ -212,7 +243,6 @@ class PanModalViewController: UIViewController, PanModalPresentable {
       (pview as! PossiblyTouchesPassableUIView).oldClass = oldClass
       (pview as! PossiblyTouchesPassableUIView).justModifiedClass = NSNumber.init(value: true)
     }
-    return self.config?.value(forKey: "shouldRoundTopCorners") as! Bool
   }
 
   var showDragIndicator: Bool {
@@ -264,13 +294,19 @@ class PanModalViewController: UIViewController, PanModalPresentable {
 
 
 extension UIViewController {
-  @objc public func presentPanModal(view: UIView, config: UIView) -> UIViewController {
-
-    let viewControllerToPresent: UIViewController & PanModalPresentable = PanModalViewController(config: config)
-    viewControllerToPresent.view = view
-    let sourceView: UIView? = nil, sourceRect: CGRect = .zero
-
-    self.presentPanModal(viewControllerToPresent, sourceView: sourceView, sourceRect: sourceRect)
-    return viewControllerToPresent
+  @objc public func presentPanModal(view: UIView, config: UIView) -> UIViewController? {
+    moved = false;
+    if self.presentedViewController != nil {
+      DispatchQueue.main.asyncAfter(deadline: .now() + 0.03) {
+        self.presentPanModal(view: view, config: config)
+      }
+    } else {
+      let viewControllerToPresent: UIViewController & PanModalPresentable = PanModalViewController(config: config)
+      viewControllerToPresent.view = view
+      let sourceView: UIView? = nil, sourceRect: CGRect = .zero
+      self.presentPanModal(viewControllerToPresent, sourceView: sourceView, sourceRect: sourceRect)
+      return viewControllerToPresent
+    }
+    return nil
   }
 }
