diff --git a/node_modules/react-native-screen-corner-radius/android/src/main/java/com/reactnativescreencornerradius/ScreenCornerRadiusModule.kt b/node_modules/react-native-screen-corner-radius/android/src/main/java/com/reactnativescreencornerradius/ScreenCornerRadiusModule.kt
index 5a20d30..7ec0064 100644
--- a/node_modules/react-native-screen-corner-radius/android/src/main/java/com/reactnativescreencornerradius/ScreenCornerRadiusModule.kt
+++ b/node_modules/react-native-screen-corner-radius/android/src/main/java/com/reactnativescreencornerradius/ScreenCornerRadiusModule.kt
@@ -11,20 +11,18 @@ import android.view.View
 import com.facebook.react.bridge.*
 
 class ScreenCornerRadiusModule(reactContext: ReactApplicationContext) :
-  ReactContextBaseJavaModule(reactContext) {
+        ReactContextBaseJavaModule(reactContext) {
   private var rootView: View? = null
 
   override fun getName(): String {
     return "ScreenCornerRadius"
   }
 
-  override fun hasConstants(): Boolean = true
-
   override fun getConstants(): MutableMap<String, Any?> {
     val cornerRadius = getCornerRadius()
 
     return mutableMapOf(
-      "cornerRadius" to cornerRadius,
+            "cornerRadius" to cornerRadius,
     )
   }
 
@@ -33,7 +31,7 @@ class ScreenCornerRadiusModule(reactContext: ReactApplicationContext) :
       rootView = currentActivity?.window?.decorView?.findViewById(android.R.id.content)
       if (rootView != null) {
         val corner =
-          rootView?.rootWindowInsets?.getRoundedCorner(RoundedCorner.POSITION_TOP_LEFT)
+                rootView?.rootWindowInsets?.getRoundedCorner(RoundedCorner.POSITION_TOP_LEFT)
         return corner?.radius ?: 0;
       }
       return 0
