diff --git a/node_modules/react-native-reanimated/src/common/style/processors/colors.ts b/node_modules/react-native-reanimated/src/common/style/processors/colors.ts
index 42d2c09..974708a 100644
--- a/node_modules/react-native-reanimated/src/common/style/processors/colors.ts
+++ b/node_modules/react-native-reanimated/src/common/style/processors/colors.ts
@@ -232,6 +232,7 @@ export function processColorsInProps(props: StyleProps) {
   for (const key in props) {
     if (!ColorProperties.includes(key)) continue;
     const value = props[key];
+    if (value == null) continue;
     props[key] = Array.isArray(value)
       ? value.map((c) => processColor(c))
       : processColor(value);
