diff --git a/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm b/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
index db88b68..2913667 100644
--- a/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
+++ b/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
@@ -28,9 +28,13 @@
 
   __block std::weak_ptr<RuntimeManager> blockRuntimeManager = runtimeManager;
 
-  dispatch_async(dispatch_get_main_queue(), ^{
-    if (blockRuntimeManager.lock()) triggerUI();
-  });
+  try {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      try {
+        if (blockRuntimeManager.lock()) triggerUI();
+      } catch (const std::exception &ex) {}
+    });
+  } catch (const std::exception &ex) {}
 }
 
 REAIOSScheduler::~REAIOSScheduler(){
diff --git a/node_modules/react-native-reanimated/src/reanimated1/derived/index.js b/node_modules/react-native-reanimated/src/reanimated1/derived/index.js
index e6e1541..943d1b6 100644
--- a/node_modules/react-native-reanimated/src/reanimated1/derived/index.js
+++ b/node_modules/react-native-reanimated/src/reanimated1/derived/index.js
@@ -2,7 +2,7 @@ export { default as acc } from './acc';
 export { default as color } from './color';
 export { default as diff } from './diff';
 export { default as diffClamp } from './diffClamp';
-export { default as interpolateNode, Extrapolate } from './interpolate';
+export { default as interpolateNode, default as interpolate, Extrapolate } from './interpolate';
 export { default as interpolateColors } from './interpolateColors';
 export { default as onChange } from './onChange';
 export { default as useCode } from './useCode';
diff --git a/node_modules/react-native-reanimated/src/reanimated1/index.js b/node_modules/react-native-reanimated/src/reanimated1/index.js
index 894c20a..a067955 100644
--- a/node_modules/react-native-reanimated/src/reanimated1/index.js
+++ b/node_modules/react-native-reanimated/src/reanimated1/index.js
@@ -35,7 +35,7 @@ export * from './derived';
 export {
   AnimatedCode as Code,
   // transitions
-  EasingNode,
+  EasingNode as Easing,
   Transitioning,
   Transition,
   createTransitioningComponent,
diff --git a/node_modules/react-native-reanimated/src/reanimated2/Colors.ts b/node_modules/react-native-reanimated/src/reanimated2/Colors.ts
index 2bc4191..38192cb 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/Colors.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/Colors.ts
@@ -9,7 +9,7 @@
 /* eslint no-bitwise: 0 */
 import { Platform } from 'react-native';
 import { makeRemote, makeShareable, isConfigured } from './core';
-import { interpolate } from './interpolation';
+import { newInterpolate as interpolate } from './interpolation';
 // @ts-ignore JS file
 import { Extrapolate } from '../reanimated1/derived';
 
diff --git a/node_modules/react-native-reanimated/src/reanimated2/Easing.ts b/node_modules/react-native-reanimated/src/reanimated2/Easing.ts
index 6fa496d..aea5300 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/Easing.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/Easing.ts
@@ -312,4 +312,4 @@ type EasingObjT = Array<keyof typeof EasingObject>;
   EasingObject[key] = createChecker(EasingObject[key], key);
 });
 
-export const Easing = EasingObject;
+export const NewEasing = EasingObject;
diff --git a/node_modules/react-native-reanimated/src/reanimated2/animations.ts b/node_modules/react-native-reanimated/src/reanimated2/animations.ts
index 9d7a35b..56df671 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/animations.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/animations.ts
@@ -1,7 +1,7 @@
 /* global _WORKLET */
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-nocheck
-import { Easing } from './Easing';
+import { NewEasing as Easing } from './Easing';
 import { isColor, convertToHSVA, toRGBA } from './Colors';
 import NativeReanimated from './NativeReanimated';
 
diff --git a/node_modules/react-native-reanimated/src/reanimated2/interpolation.ts b/node_modules/react-native-reanimated/src/reanimated2/interpolation.ts
index 860af01..39a8319 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/interpolation.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/interpolation.ts
@@ -113,7 +113,7 @@ function internalInterpolate(x, l, r, ll, rr, type) {
   return val;
 }
 
-export function interpolate(x, input, output, type) {
+export function newInterpolate(x, input, output, type) {
   'worklet';
   if (x && x.__nodeID) {
     console.warn(
