diff --git a/node_modules/react-navigation-stack/src/views/KeyboardManager.tsx b/node_modules/react-navigation-stack/src/views/KeyboardManager.tsx
index 01721bb..571fc5d 100644
--- a/node_modules/react-navigation-stack/src/views/KeyboardManager.tsx
+++ b/node_modules/react-navigation-stack/src/views/KeyboardManager.tsx
@@ -15,10 +15,13 @@ export default class KeyboardManager extends React.Component<Props> {
   private previouslyFocusedTextInput: number | null = null;

   private handlePageChangeStart = () => {
-    const input = TextInput.State.currentlyFocusedField();
+    const input = TextInput.State.currentlyFocusedField()
+      || this.previouslyFocusedTextInput;

     // When a page change begins, blur the currently focused input
-    TextInput.State.blurTextInput(input);
+    if (input !== null) {
+      TextInput.State.blurTextInput(input);
+    }

     // Store the id of this input so we can refocus it if change was cancelled
     this.previouslyFocusedTextInput = input;
@@ -38,8 +41,6 @@ export default class KeyboardManager extends React.Component<Props> {
     if (input) {
       TextInput.State.focusTextInput(input);
     }
-
-    this.previouslyFocusedTextInput = null;
   };

   render() {