diff --git a/node_modules/react-native-bignumber/lib/commonjs/index.js b/node_modules/react-native-bignumber/lib/commonjs/index.js
index cf9008a..ea08631 100644
--- a/node_modules/react-native-bignumber/lib/commonjs/index.js
+++ b/node_modules/react-native-bignumber/lib/commonjs/index.js
@@ -1,25 +1,7 @@
-"use strict";
 
-Object.defineProperty(exports, "__esModule", {
-  value: true
-});
+if (global.storage) {
+  throw new Error("commonjs rn-bignumber lib used in app. This should be only for tests!!")
+}
 
-var _BigNumber = _interopRequireWildcard(require("./BigNumber"));
-
-Object.keys(_BigNumber).forEach(function (key) {
-  if (key === "default" || key === "__esModule") return;
-  if (key in exports && exports[key] === _BigNumber[key]) return;
-  Object.defineProperty(exports, key, {
-    enumerable: true,
-    get: function () {
-      return _BigNumber[key];
-    }
-  });
-});
-
-function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
-
-function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
-
-module.exports = _BigNumber.default;
+module.exports = require("bn.js")
 //# sourceMappingURL=index.js.map
