diff --git a/node_modules/react-native-keyboard-area/android/build.gradle b/node_modules/react-native-keyboard-area/android/build.gradle
index 4fc21b5..467828f 100644
--- a/node_modules/react-native-keyboard-area/android/build.gradle
+++ b/node_modules/react-native-keyboard-area/android/build.gradle
@@ -1,6 +1,6 @@
 buildscript {
     ext {
-        kotlinVersion = '1.6.0'
+        kotlinVersion = '1.7.21'
         androidToolsVersion = '3.2.1'
     }
 
