diff --git a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.java b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.java
index 709727b..464cf12 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.java
+++ b/node_modules/react-native-gesture-handler/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.java
@@ -42,7 +42,8 @@ public class RNGestureHandlerButtonViewManager extends
     boolean mUseBorderless = false;
     float mBorderRadius = 0;
     boolean mNeedBackgroundUpdate = false;
-    long mLastEventTime = 0;
+    long mLastEventTime = -1;
+    int mLastAction = -1;
     public static final String SELECTABLE_ITEM_BACKGROUND = "selectableItemBackground";
     public static final String SELECTABLE_ITEM_BACKGROUND_BORDERLESS = "selectableItemBackgroundBorderless";
 
@@ -127,8 +128,11 @@ public class RNGestureHandlerButtonViewManager extends
     @Override
     public boolean onTouchEvent(MotionEvent event) {
       long eventTime = event.getEventTime();
-      if (mLastEventTime != eventTime || mLastEventTime == 0) {
+      // https://github.com/software-mansion/react-native-gesture-handler/pull/1512
+      int action = event.getAction();
+      if (mLastEventTime != eventTime || mLastAction != action) {
         mLastEventTime = eventTime;
+        mLastAction = action;
         return super.onTouchEvent(event);
       }
       return false;
diff --git a/node_modules/react-native-gesture-handler/ios/Handlers/RNNativeViewHandler.m b/node_modules/react-native-gesture-handler/ios/Handlers/RNNativeViewHandler.m
index 73022d7..49b8507 100644
--- a/node_modules/react-native-gesture-handler/ios/Handlers/RNNativeViewHandler.m
+++ b/node_modules/react-native-gesture-handler/ios/Handlers/RNNativeViewHandler.m
@@ -78,7 +78,7 @@
         // we expect for RCTScrollView component to contain a subclass of UIScrollview as the only
         // subview
         UIScrollView *scrollView = [view.subviews objectAtIndex:0];
-        scrollView.delaysContentTouches = YES;
+        scrollView.delaysContentTouches = NO;
     }
 }
 
diff --git a/node_modules/react-native-gesture-handler/src/PlatformConstants.ts b/node_modules/react-native-gesture-handler/src/PlatformConstants.ts
index cf7ad64..4ef5cdf 100644
--- a/node_modules/react-native-gesture-handler/src/PlatformConstants.ts
+++ b/node_modules/react-native-gesture-handler/src/PlatformConstants.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import { NativeModules, Platform } from 'react-native';
 
 type PlatformConstants = {
diff --git a/node_modules/react-native-gesture-handler/src/components/GestureButtons.tsx b/node_modules/react-native-gesture-handler/src/components/GestureButtons.tsx
index 9dc1439..88303c1 100644
--- a/node_modules/react-native-gesture-handler/src/components/GestureButtons.tsx
+++ b/node_modules/react-native-gesture-handler/src/components/GestureButtons.tsx
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import * as React from 'react';
 import {
   Animated,
diff --git a/node_modules/react-native-gesture-handler/src/handlers/createHandler.ts b/node_modules/react-native-gesture-handler/src/handlers/createHandler.ts
index 38b32ea..0b70114 100644
--- a/node_modules/react-native-gesture-handler/src/handlers/createHandler.ts
+++ b/node_modules/react-native-gesture-handler/src/handlers/createHandler.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import * as React from 'react';
 import {
   findNodeHandle as findNodeHandleRN,
diff --git a/node_modules/react-native-gesture-handler/src/handlers/createNativeWrapper.tsx b/node_modules/react-native-gesture-handler/src/handlers/createNativeWrapper.tsx
index f90b23b..6e8b054 100644
--- a/node_modules/react-native-gesture-handler/src/handlers/createNativeWrapper.tsx
+++ b/node_modules/react-native-gesture-handler/src/handlers/createNativeWrapper.tsx
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import * as React from 'react';
 import { useImperativeHandle, useRef } from 'react';
 
diff --git a/node_modules/react-native-gesture-handler/src/handlers/gestureHandlers.ts b/node_modules/react-native-gesture-handler/src/handlers/gestureHandlers.ts
index d2be2b4..a68916d 100644
--- a/node_modules/react-native-gesture-handler/src/handlers/gestureHandlers.ts
+++ b/node_modules/react-native-gesture-handler/src/handlers/gestureHandlers.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 // Previous types exported gesture handlers as classes which creates an interface and variable, both named the same as class.
 // Without those types, we'd introduce breaking change, forcing users to prefix every handler type specification with typeof
 // e.g. React.createRef<TapGestureHandler> -> React.createRef<typeof TapGestureHandler>.
diff --git a/node_modules/react-native-gesture-handler/src/web/GestureHandler.ts b/node_modules/react-native-gesture-handler/src/web/GestureHandler.ts
index 5f3f9f4..369171e 100644
--- a/node_modules/react-native-gesture-handler/src/web/GestureHandler.ts
+++ b/node_modules/react-native-gesture-handler/src/web/GestureHandler.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 /* eslint-disable eslint-comments/no-unlimited-disable */
 /* eslint-disable */
 import Hammer from '@egjs/hammerjs';
diff --git a/node_modules/react-native-gesture-handler/src/web/PanGestureHandler.ts b/node_modules/react-native-gesture-handler/src/web/PanGestureHandler.ts
index b588d69..6d7c466 100644
--- a/node_modules/react-native-gesture-handler/src/web/PanGestureHandler.ts
+++ b/node_modules/react-native-gesture-handler/src/web/PanGestureHandler.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import Hammer from '@egjs/hammerjs';
 
 import {
diff --git a/node_modules/react-native-gesture-handler/src/web/TapGestureHandler.ts b/node_modules/react-native-gesture-handler/src/web/TapGestureHandler.ts
index 312049c..0562f9c 100644
--- a/node_modules/react-native-gesture-handler/src/web/TapGestureHandler.ts
+++ b/node_modules/react-native-gesture-handler/src/web/TapGestureHandler.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import Hammer from '@egjs/hammerjs';
 
 import DiscreteGestureHandler from './DiscreteGestureHandler';