diff --git a/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m b/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
index 637f112..3374bd0 100644
--- a/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
+++ b/node_modules/react-native-slack-bottom-sheet/ios/SlackBottomSheet.m
@@ -39,6 +39,7 @@ - (void)layoutSubviews {
 @end

 @interface InvisibleView: RCTView
+- (void)jumpTo:(NSNumber*)longForm;
 @property (nonatomic, nonnull) NSNumber *topOffset;
 @property (nonatomic) BOOL isShortFormEnabled;
 @property (nonatomic, nullable) NSNumber *longFormHeight;
@@ -54,6 +55,7 @@ @interface InvisibleView: RCTView
 @property (nonatomic) BOOL isHapticFeedbackEnabled;
 @property (nonatomic) BOOL shouldRoundTopCorners;
 @property (nonatomic) BOOL showDragIndicator;
+@property (nonatomic) BOOL gestureEnabled;
 @property (nonatomic) BOOL blocksBackgroundTouches;
 @property (nonatomic) BOOL interactsWithOuterScrollView;
 @property (nonatomic) BOOL presentGlobally;
@@ -66,6 +68,7 @@ @interface InvisibleView: RCTView
 @property (nonatomic, copy, nullable) RCTBubblingEventBlock onWillTransition;
 @property (nonatomic, copy, nullable) RCTBubblingEventBlock onWillDismiss;
 @property (nonatomic, copy, nullable) RCTBubblingEventBlock onDidDismiss;
+@property (nonatomic, copy, nullable) RCTBubblingEventBlock onCrossMagicBorder;
 @end

 @implementation InvisibleView {
@@ -109,6 +112,7 @@ - (instancetype)initWithBridge:(RCTBridge *)bridge {
     _modalPresented = false;
     _scrollsToTop = false;
     _isHiding = true;
+    _gestureEnabled = true;

   }
   return self;
@@ -121,6 +125,19 @@ -(void) setScrollsToTopOnTapStatusBar:(BOOL) scrollsToTop {
   }
 }

+-(void)jumpTo:(NSNumber*)longForm {
+  if (_contoller != nil) {
+    [_contoller performSelector:NSSelectorFromString(@"jumpToLong:") withObject:longForm];
+  }
+}
+
+
+-(void)layout {
+  if (_contoller != nil) {
+    [_contoller performSelector:NSSelectorFromString(@"panModalSetNeedsLayoutUpdateWrapper")];
+  }
+}
+
 -(void)reactSetFrame:(CGRect)frame {
   // shrug
 }
@@ -147,6 +164,18 @@ - (void)callDidDismiss {
   _onDidDismiss(@{});
 }

+- (void)setGestureEnabled:(BOOL)gestureEnabled {
+  [_contoller.view.superview.superview.gestureRecognizers[0] setEnabled:gestureEnabled];
+}
+
+- (void)callOnCrossMagicBoderFromTop {
+  _onCrossMagicBorder(@{@"below": @NO});
+}
+
+- (void)callOnCrossMagicBoderFromBottom {
+  _onCrossMagicBorder(@{@"below": @YES});
+}
+
 - (void)callWillTransitionLong {
   _onWillTransition(@{@"type": @"long"});
 }
@@ -182,7 +211,7 @@ - (void)setVisible:(BOOL)visible {
       self->_modalPresented = YES;
     });
   } else {
-    RCTExecuteOnMainQueue(^{
+    //RCTExecuteOnMainQueue(^{
       if (!self->_modalPresented) {
         return;
       }
@@ -193,12 +222,17 @@ - (void)setVisible:(BOOL)visible {
       self.transitionDuration = [[NSNumber alloc] initWithDouble: 0];
       UIViewController *rootViewController = [UIApplication sharedApplication].delegate.window.rootViewController;

-      [[rootViewController presentedViewController] dismissViewControllerAnimated:!self->_isHiding completion:^{
+      [[rootViewController presentedViewController] dismissViewControllerAnimated:NO completion:^{
+        if (self.onDidDismiss) {
+          self.onDidDismiss(nil);
+        }
         self.transitionDuration = oldTransitionDuration;
       }];
+      //self->_contoller.view = nil;
+      self->_contoller = nil;
       self->_isHiding = false;
       self->_modalPresented = NO;
-    });
+    //});
   }
 }

@@ -213,7 +247,7 @@ - (void)addSubview:(UIView *)view {

 @end

-@interface ModalViewManager : RCTViewManager
+@interface ModalViewManager : RCTViewManager<RCTBridgeModule>
 @end

 @implementation ModalViewManager
@@ -234,12 +268,14 @@ @implementation ModalViewManager
 RCT_EXPORT_VIEW_PROPERTY(allowsDragToDismiss, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(isUserInteractionEnabled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(blocksBackgroundTouches, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(gestureEnabled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(isHapticFeedbackEnabled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(shouldRoundTopCorners, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(showDragIndicator, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(startFromShortForm, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(onWillTransition, RCTBubblingEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(onDidDismiss, RCTBubblingEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onCrossMagicBorder, RCTBubblingEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(presentGlobally, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(interactsWithOuterScrollView, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(scrollsToTopOnTapStatusBar, BOOL)
@@ -247,6 +283,23 @@ @implementation ModalViewManager
 RCT_EXPORT_VIEW_PROPERTY(unmountAnimation, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(visible, BOOL)

+RCT_EXPORT_METHOD(jumpTo:(nonnull NSNumber*)point tag:(nonnull NSNumber*) reactTag) {
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *,UIView *> *viewRegistry) {
+    InvisibleView *view = (InvisibleView *)viewRegistry[reactTag];
+    [view jumpTo:point];
+  }];
+
+}
+
+RCT_EXPORT_METHOD(layout:(nonnull NSNumber*) reactTag) {
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *,UIView *> *viewRegistry) {
+    InvisibleView *view = (InvisibleView *)viewRegistry[reactTag];
+    [view layout];
+  }];
+
+}
+
+
 - (UIView *)view {
   return [[InvisibleView alloc] initWithBridge:self.bridge];
 }
diff --git a/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift b/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
index b688639..14af71f 100644
--- a/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
+++ b/node_modules/react-native-slack-bottom-sheet/ios/UIViewController+PanModalPresenter.swift
@@ -1,6 +1,7 @@
 import UIKit

 import PanModal
+import React

 class BetterGestureRecognizerDelegateAdapter: NSObject, UIGestureRecognizerDelegate {
   var grd: UIGestureRecognizerDelegate
@@ -29,6 +30,8 @@ class BetterGestureRecognizerDelegateAdapter: NSObject, UIGestureRecognizerDeleg
   }
 }

+var moved = false
+
 class PossiblyTouchesPassableUIView: UIView {
   var grdelegate: UIGestureRecognizerDelegate?
   var config: NSObject?
@@ -49,12 +52,17 @@ class PossiblyTouchesPassableUIView: UIView {
     super.layoutSubviews()
     let outerView = self.config?.value(forKey: "outerView") as? UIView
     if (!(self.config!.value(forKey: "presentGlobally") as! Bool)) {
+      if moved {
+        return
+      }
+      moved = true
       removeFromSuperview()
       let helperView: UIView = self.subviews[1].subviews[0]
-      let bounds = outerView!.bounds
-      let topOffset:CGFloat = (topLayoutGuideLength ?? 0) + CGFloat(truncating: self.config?.value(forKey: "topOffset") as! NSNumber)
-      let newBounds = CGRect.init(x: bounds.minX, y: bounds.minY, width: bounds.width, height: bounds.height - topOffset)
-      helperView.setValue(newBounds, forKeyPath: "specialBounds")
+        let bounds = outerView!.bounds
+        let topOffset:CGFloat = (topLayoutGuideLength ?? 0) + CGFloat(truncating: self.config?.value(forKey: "topOffset") as! NSNumber)
+
+        let newBounds = CGRect.init(x: bounds.minX, y: bounds.minY, width: bounds.width, height: bounds.height - topOffset)
+        helperView.setValue(newBounds, forKeyPath: "specialBounds")
       outerView?.addSubview(self)
     }
     let gr: UIGestureRecognizer = self.gestureRecognizers![0]
@@ -97,6 +105,18 @@ class PanModalViewController: UIViewController, PanModalPresentable {

   }

+  var observation: NSKeyValueObservation?
+
+  @objc func jumpTo(long: NSNumber) {
+    self.panModalSetNeedsLayoutUpdate()
+    if (long.boolValue) {
+      panModalTransition(to: .longForm);
+    } else {
+      panModalTransition(to: .shortForm);
+    }
+    self.panModalSetNeedsLayoutUpdate()
+  }
+
   var config: NSObject?
   var scrollView: UIScrollView?
   convenience init(config: NSObject) {
@@ -104,6 +124,14 @@ class PanModalViewController: UIViewController, PanModalPresentable {
     self.config = config
   }

+  override func viewDidLoad() {
+    super.viewDidLoad()
+    hack()
+  }
+
+  @objc func panModalSetNeedsLayoutUpdateWrapper() {
+    panModalSetNeedsLayoutUpdate()
+  }

   override var preferredStatusBarStyle: UIStatusBarStyle {
     return .lightContent
@@ -203,6 +231,25 @@ class PanModalViewController: UIViewController, PanModalPresentable {
     return self.config?.value(forKey: "isHapticFeedbackEnabled") as! Bool  }

   var shouldRoundTopCorners: Bool {
+    hack()
+    return self.config?.value(forKey: "shouldRoundTopCorners") as! Bool
+  }
+
+  override func viewDidLayoutSubviews() {
+    hack()
+  }
+
+  func updatePostion(position: CGFloat, newPostion: CGFloat) {
+    if (position < 80.0 && newPostion >= 80.0) {
+      self.config?.performSelector(inBackground: Selector.init(("callOnCrossMagicBoderFromBottom")), with: nil)
+    }
+
+    if (position >= 80.0 && newPostion < 80.0) {
+      self.config?.performSelector(inBackground: Selector.init(("callOnCrossMagicBoderFromTop")), with: nil)
+    }
+  }
+
+  public func hack() {
     let pview = view.superview!.superview!
     if !(pview is PossiblyTouchesPassableUIView) {
       let oldClass: AnyClass = type(of: pview)
@@ -211,8 +258,10 @@ class PanModalViewController: UIViewController, PanModalPresentable {
       (pview as! PossiblyTouchesPassableUIView).topLayoutGuideLength = self.topLayoutGuide.length
       (pview as! PossiblyTouchesPassableUIView).oldClass = oldClass
       (pview as! PossiblyTouchesPassableUIView).justModifiedClass = NSNumber.init(value: true)
+      observation = (self.presentationController as! PanModalPresentationController).observe(\.yPosition, options: [.old, .new]) { object, change in
+        self.updatePostion(position: change.oldValue!, newPostion: change.newValue!)
+      }
     }
-    return self.config?.value(forKey: "shouldRoundTopCorners") as! Bool
   }

   var showDragIndicator: Bool {
@@ -264,13 +313,19 @@ class PanModalViewController: UIViewController, PanModalPresentable {


 extension UIViewController {
-  @objc public func presentPanModal(view: UIView, config: UIView) -> UIViewController {
-
-    let viewControllerToPresent: UIViewController & PanModalPresentable = PanModalViewController(config: config)
-    viewControllerToPresent.view = view
-    let sourceView: UIView? = nil, sourceRect: CGRect = .zero
-
-    self.presentPanModal(viewControllerToPresent, sourceView: sourceView, sourceRect: sourceRect)
-    return viewControllerToPresent
+  @objc public func presentPanModal(view: UIView, config: UIView) -> UIViewController? {
+    moved = false;
+    if self.presentedViewController != nil {
+      DispatchQueue.main.asyncAfter(deadline: .now() + 0.03) {
+        self.presentPanModal(view: view, config: config)
+      }
+    } else {
+      let viewControllerToPresent: UIViewController & PanModalPresentable = PanModalViewController(config: config)
+      viewControllerToPresent.view = view
+      let sourceView: UIView? = nil, sourceRect: CGRect = .zero
+      self.presentPanModal(viewControllerToPresent, sourceView: sourceView, sourceRect: sourceRect)
+      return viewControllerToPresent
+    }
+    return nil
   }
 }
