diff --git a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
index 7101410..272736e 100644
--- a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
+++ b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
@@ -357,7 +357,7 @@ - (BOOL)cleanTmpDirectory {
                 }
             }
             
-            [imagePickerController setModalPresentationStyle: UIModalPresentationFullScreen];
+            [imagePickerController setModalPresentationStyle: UIModalPresentationOverFullScreen];
             [[self getRootVC] presentViewController:imagePickerController animated:YES completion:nil];
         });
     }];
