diff --git a/node_modules/react-native-dark-mode/android/build.gradle b/node_modules/react-native-dark-mode/android/build.gradle
index 11c2a43..263ef0c 100644
--- a/node_modules/react-native-dark-mode/android/build.gradle
+++ b/node_modules/react-native-dark-mode/android/build.gradle
@@ -14 +14 @@
-			jcenter()
+			mavenCentral()
@@ -23 +23 @@
-apply plugin: 'maven'
+// apply plugin: 'maven'
@@ -58 +58 @@
-	jcenter()
+	mavenCentral()
