diff --git a/node_modules/react-native-fast-image/dist/index.d.ts b/node_modules/react-native-fast-image/dist/index.d.ts
index 8a91257..fd05060 100644
--- a/node_modules/react-native-fast-image/dist/index.d.ts
+++ b/node_modules/react-native-fast-image/dist/index.d.ts
@@ -54,7 +54,7 @@ export interface ImageStyle extends FlexStyle, TransformsStyle, ShadowStyleIOS {
     opacity?: number;
 }
 export interface FastImageProps extends AccessibilityProps, ViewProps {
-    source: Source | number;
+    source: Source | number | StaticImageData;
     resizeMode?: ResizeMode;
     fallback?: boolean;
     onLoadStart?(): void;
@@ -95,6 +95,7 @@ export interface FastImageStaticProperties {
     priority: typeof priority;
     cacheControl: typeof cacheControl;
     preload: (sources: Source[]) => void;
+    getCachePath: (source: Source) => Promise<string>
     clearMemoryCache: () => Promise<void>;
     clearDiskCache: () => Promise<void>;
 }
