diff --git a/node_modules/react-native-plaid-link-sdk/android/build.gradle b/node_modules/react-native-plaid-link-sdk/android/build.gradle
index c1c0b7d..4e720d1 100644
--- a/node_modules/react-native-plaid-link-sdk/android/build.gradle
+++ b/node_modules/react-native-plaid-link-sdk/android/build.gradle
@@ -12,7 +12,7 @@ allprojects {
 
 
 buildscript {
-    ext.kotlin_version = '1.3.61'
+    ext.kotlin_version = '1.7.21'
     repositories {
         google()
         mavenCentral()
