diff --git a/node_modules/react-native-sound/RNSound/RNSound.m b/node_modules/react-native-sound/RNSound/RNSound.m
index 5950dfe..339a14d 100644
--- a/node_modules/react-native-sound/RNSound/RNSound.m
+++ b/node_modules/react-native-sound/RNSound/RNSound.m
@@ -195,19 +195,19 @@ - (NSDictionary *)constantsToExport {
     NSURL *fileNameUrl;
     AVAudioPlayer *player;
 
-    if ([fileName hasPrefix:@"http"]) {
-        fileNameUrl = [NSURL URLWithString:fileName];
-        NSData *data = [NSData dataWithContentsOfURL:fileNameUrl];
-        player = [[AVAudioPlayer alloc] initWithData:data error:&error];
-    } else if ([fileName hasPrefix:@"ipod-library://"]) {
-        fileNameUrl = [NSURL URLWithString:fileName];
-        player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileNameUrl
-                                                        error:&error];
-    } else {
+    //if ([fileName hasPrefix:@"http"]) {
+    //    fileNameUrl = [NSURL URLWithString:fileName];
+    //    NSData *data = [NSData dataWithContentsOfURL:fileNameUrl];
+    //    player = [[AVAudioPlayer alloc] initWithData:data error:&error];
+    //} else if ([fileName hasPrefix:@"ipod-library://"]) {
+    //    fileNameUrl = [NSURL URLWithString:fileName];
+    //    player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileNameUrl
+    //                                                    error:&error];
+    //} else {
         fileNameUrl = [NSURL URLWithString:fileName];
         player = [[AVAudioPlayer alloc] initWithContentsOfURL:fileNameUrl
                                                         error:&error];
-    }
+    //}
 
     if (player) {
         @synchronized(self) {
