diff --git a/node_modules/react-native-sound/sound.js b/node_modules/react-native-sound/sound.js
index 015cf43..19bbc4f 100644
--- a/node_modules/react-native-sound/sound.js
+++ b/node_modules/react-native-sound/sound.js
@@ -1,11 +1,11 @@
 'use strict';
 
-var ReactNative = require('react-native');
-var RNSound = ReactNative.NativeModules.RNSound;
+import { NativeEventEmitter, NativeModules } from 'react-native';
+const { RNSound } = NativeModules;
 var IsAndroid = RNSound.IsAndroid;
 var IsWindows = RNSound.IsWindows;
 var resolveAssetSource = require("react-native/Libraries/Image/resolveAssetSource");
-var eventEmitter = new ReactNative.NativeEventEmitter(RNSound);
+var eventEmitter = new NativeEventEmitter(RNSound);
 
 var nextKey = 0;
 
@@ -277,4 +277,5 @@ Sound.DOCUMENT = RNSound.NSDocumentDirectory;
 Sound.LIBRARY = RNSound.NSLibraryDirectory;
 Sound.CACHES = RNSound.NSCachesDirectory;
 
-module.exports = Sound;
+export default Sound;
+
