diff --git a/node_modules/react-native-redash/src/v1/Gesture.ts b/node_modules/react-native-redash/src/v1/Gesture.ts
index 47c83ab..f5d4472 100644
--- a/node_modules/react-native-redash/src/v1/Gesture.ts
+++ b/node_modules/react-native-redash/src/v1/Gesture.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import Animated, { diff, lessThan, or } from "react-native-reanimated";
 import type {
   FlingGestureHandlerEventExtra,
diff --git a/node_modules/react-native-redash/src/v1/ReText.tsx b/node_modules/react-native-redash/src/v1/ReText.tsx
index c395935..9f772a4 100644
--- a/node_modules/react-native-redash/src/v1/ReText.tsx
+++ b/node_modules/react-native-redash/src/v1/ReText.tsx
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import * as React from "react";
 import type { TextStyle, StyleProp } from "react-native";
 import { TextInput } from "react-native";
diff --git a/node_modules/react-native-redash/src/v1/SVG.ts b/node_modules/react-native-redash/src/v1/SVG.ts
index 9751193..81ec8d8 100644
--- a/node_modules/react-native-redash/src/v1/SVG.ts
+++ b/node_modules/react-native-redash/src/v1/SVG.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import Animated, { interpolateNode } from "react-native-reanimated";
 import parseSVG from "parse-svg-path";
 import absSVG from "abs-svg-path";
