diff --git a/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts b/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
index bbcd336..8e962f1 100644
--- a/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
+++ b/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
@@ -1,4 +1,4 @@
-declare const AnimateableText: import("react").FunctionComponent<import("react-native-reanimated").AnimateProps<Omit<import("react-native").TextProps, "children"> & {
+declare const AnimateableText: import("react").FunctionComponent<import("react-native-reanimated").AnimateProps<import("react-native").TextProps & {
     forwardedRef?: import("react").Ref<import("react-native").Text>;
     text?: string;
     onStartShouldSetResponder?: () => boolean;
diff --git a/node_modules/react-native-animateable-text/src/TextProps.tsx b/node_modules/react-native-animateable-text/src/TextProps.tsx
index 62f84eb..9f9a046 100644
--- a/node_modules/react-native-animateable-text/src/TextProps.tsx
+++ b/node_modules/react-native-animateable-text/src/TextProps.tsx
@@ -5,7 +5,7 @@ import type {
   GestureResponderEvent,
 } from "react-native";
 
-export type AnimateableTextProps = Omit<NativeTextProps, "children"> & {
+export type AnimateableTextProps = NativeTextProps & {
   forwardedRef?: Ref<IText>;
   text?: string;
 
