diff --git a/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextShadowNode.java b/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextShadowNode.java
index 95baadc..dbab5a3 100644
--- a/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextShadowNode.java
+++ b/node_modules/react-native-animateable-text/android/src/main/java/com/reactnativereanimatedtext/JBTextShadowNode.java
@@ -6,27 +6,14 @@ package com.reactnativereanimatedtext;
 
 import android.annotation.TargetApi;
 import android.os.Build;
-import com.facebook.react.uimanager.NativeViewHierarchyOptimizer;
 import com.facebook.react.uimanager.annotations.ReactProp;
 import com.facebook.react.views.text.ReactTextShadowNode;
 import com.facebook.react.views.text.ReactTextViewManagerCallback;
 
-import java.lang.reflect.Field;
-
 @TargetApi(Build.VERSION_CODES.M)
 public class JBTextShadowNode extends ReactTextShadowNode {
   protected String mText = "";
 
-private static final Field mPreparedSpannableTextField;
-  static {
-    try {
-      mPreparedSpannableTextField = ReactTextShadowNode.class.getDeclaredField("mPreparedSpannableText");
-      mPreparedSpannableTextField.setAccessible(true);
-    } catch (NoSuchFieldException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
   public JBTextShadowNode() {
     this(null);
   }
@@ -35,24 +22,6 @@ private static final Field mPreparedSpannableTextField;
     super(reactTextViewManagerCallback);
   }
 
-  @Override
-  public void onBeforeLayout(NativeViewHierarchyOptimizer nativeViewHierarchyOptimizer) {
-    // ADDED
-    try {
-    // END ADDED
-      mPreparedSpannableTextField.set(this, spannedFromShadowNode(
-        this,
-        /* text (e.g. from `value` prop): */ mText, // EDITED
-        /* supportsInlineViews: */ true,
-        nativeViewHierarchyOptimizer));
-      markUpdated();
-    // ADDED
-    } catch (IllegalAccessException e) {
-      throw new RuntimeException(e);
-    }
-    // END ADDED
-  }
-
   @ReactProp(name = "text")
   public void setText(String newText) {
     mText = newText;
diff --git a/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts b/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
index bbcd336..9b726a9 100644
--- a/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
+++ b/node_modules/react-native-animateable-text/lib/typescript/src/index.d.ts
@@ -1,4 +1,4 @@
-declare const AnimateableText: import("react").FunctionComponent<import("react-native-reanimated").AnimateProps<Omit<import("react-native").TextProps, "children"> & {
+declare const AnimateableText: import("react").FunctionComponent<import("react-native-reanimated").AnimateProps<import("react-native").TextProps & {
     forwardedRef?: import("react").Ref<import("react-native").Text>;
     text?: string;
     onStartShouldSetResponder?: () => boolean;
diff --git a/node_modules/react-native-animateable-text/src/TextProps.tsx b/node_modules/react-native-animateable-text/src/TextProps.tsx
index 62f84eb..9f9a046 100644
--- a/node_modules/react-native-animateable-text/src/TextProps.tsx
+++ b/node_modules/react-native-animateable-text/src/TextProps.tsx
@@ -5,7 +5,7 @@ import type {
   GestureResponderEvent,
 } from "react-native";
 
-export type AnimateableTextProps = Omit<NativeTextProps, "children"> & {
+export type AnimateableTextProps = NativeTextProps & {
   forwardedRef?: Ref<IText>;
   text?: string;
 
