diff --git a/node_modules/react-native-safe-area-view/index.js b/node_modules/react-native-safe-area-view/index.js
index ac1b6d3..f17bd76 100644
--- a/node_modules/react-native-safe-area-view/index.js
+++ b/node_modules/react-native-safe-area-view/index.js
@@ -27,7 +27,7 @@ const PAD_HEIGHT = 1024;
 
 const { height: D_HEIGHT, width: D_WIDTH } = Dimensions.get('window');
 
-const { PlatformConstants = {} } = NativeModules;
+const PlatformConstants = Platform.constants || {};
 const { minor = 0 } = PlatformConstants.reactNativeVersion || {};
 
 const isIPhoneX = (() => {
