diff --git a/node_modules/react-native-tab-view/lib/module/Pager.js b/node_modules/react-native-tab-view/lib/module/Pager.js
index d3e8a5b..6b9391b 100644
--- a/node_modules/react-native-tab-view/lib/module/Pager.js
+++ b/node_modules/react-native-tab-view/lib/module/Pager.js
@@ -64,8 +64,8 @@ const SPRING_CONFIG = {
 };
 const SPRING_VELOCITY_SCALE = 1;
 const TIMING_CONFIG = {
-  duration: 200,
-  easing: Easing.out(Easing.cubic)
+  duration: 275,
+  easing: Easing.bezier(0.22, 1, 0.36, 1)
 };
 export default class Pager extends React.Component {
   constructor(...args) {
