diff --git a/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm b/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
index 541c4fa..4be2e7c 100644
--- a/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
+++ b/node_modules/react-native-reanimated/ios/native/REAIOSScheduler.mm
@@ -26,10 +26,15 @@
   }
   
   __block std::weak_ptr<NativeReanimatedModule> blockModule = module;
-  
-  dispatch_async(dispatch_get_main_queue(), ^{
-    if (blockModule.lock()) triggerUI();
-  });
+  try {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      try {
+        if (module.lock()) {
+          triggerUI();
+        }
+      } catch (const std::exception &ex) {}
+    });
+  } catch (const std::exception &ex) {}
 }
 
 REAIOSScheduler::~REAIOSScheduler(){
diff --git a/node_modules/react-native-reanimated/src/Animated.js b/node_modules/react-native-reanimated/src/Animated.js
index 523853a..d7813c3 100644
--- a/node_modules/react-native-reanimated/src/Animated.js
+++ b/node_modules/react-native-reanimated/src/Animated.js
@@ -81,7 +81,7 @@ export * from './derived';
 export * from './reanimated2';

 export {
-  EasingNode,
+  EasingNode as Easing,
   Transitioning,
   Transition,
   createTransitioningComponent,
diff --git a/node_modules/react-native-reanimated/src/derived/index.js b/node_modules/react-native-reanimated/src/derived/index.js
index e6e1541..943d1b6 100644
--- a/node_modules/react-native-reanimated/src/derived/index.js
+++ b/node_modules/react-native-reanimated/src/derived/index.js
@@ -2,7 +2,7 @@ export { default as acc } from './acc';
 export { default as color } from './color';
 export { default as diff } from './diff';
 export { default as diffClamp } from './diffClamp';
-export { default as interpolateNode, Extrapolate } from './interpolate';
+export { default as interpolateNode, default as interpolate, Extrapolate } from './interpolate';
 export { default as interpolateColors } from './interpolateColors';
 export { default as onChange } from './onChange';
 export { default as useCode } from './useCode';
diff --git a/node_modules/react-native-reanimated/src/reanimated2/Colors.js b/node_modules/react-native-reanimated/src/reanimated2/Colors.js
index 2fa2578..a095883 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/Colors.js
+++ b/node_modules/react-native-reanimated/src/reanimated2/Colors.js
@@ -10,7 +10,7 @@

 import { Platform } from 'react-native';
 import { makeRemote, makeShareable } from './core';
-import { interpolate } from './interpolation';
+import { newInterpolate as interpolate } from './interpolation';
 import { Extrapolate } from '../derived';

 // var INTEGER = '[-+]?\\d+';
diff --git a/node_modules/react-native-reanimated/src/reanimated2/Easing.js b/node_modules/react-native-reanimated/src/reanimated2/Easing.js
index 730f366..c3890c1 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/Easing.js
+++ b/node_modules/react-native-reanimated/src/reanimated2/Easing.js
@@ -288,5 +288,5 @@ Object.keys(EasingObject).forEach((key) => {
   EasingObject[key] = createChecker(EasingObject[key], key);
 });

-export const Easing = EasingObject;
-export default Easing;
+export const NewEasing = EasingObject;
+export default NewEasing;
diff --git a/node_modules/react-native-reanimated/src/reanimated2/animations.js b/node_modules/react-native-reanimated/src/reanimated2/animations.js
index 3fc07ea..221fee3 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/animations.js
+++ b/node_modules/react-native-reanimated/src/reanimated2/animations.js
@@ -1,5 +1,5 @@
 /* global _WORKLET */
-import { Easing } from './Easing';
+import { NewEasing as Easing } from './Easing';
 import { isColor, convertToHSVA, toRGBA } from './Colors';
 import NativeReanimated from './NativeReanimated';

@@ -372,7 +372,7 @@ export function withDecay(userConfig, callback) {

       const v0 = velocity / 1000;
       const v = v0 * kv * 1000;
-      const x = current + v0 * kx;
+      const x = current + v0 * (config.deceleration === 1 ? 1 : kx);

       animation.current = x;
       animation.velocity = v;
diff --git a/node_modules/react-native-reanimated/src/reanimated2/interpolation.js b/node_modules/react-native-reanimated/src/reanimated2/interpolation.js
index 28a96f9..97a1144 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/interpolation.js
+++ b/node_modules/react-native-reanimated/src/reanimated2/interpolation.js
@@ -111,7 +111,7 @@ function internalInterpolate(x, l, r, ll, rr, type) {
   return val;
 }

-export function interpolate(x, input, output, type) {
+export function newInterpolate(x, input, output, type) {
   'worklet';
   if (x && x.__nodeID) {
     console.warn(`interpolate() was renamed to interpolateNode() in Reanimated 2. Please use interpolateNode() instead`);
