diff --git a/node_modules/react-native-change-icon/ios/ChangeIcon.swift b/node_modules/react-native-change-icon/ios/ChangeIcon.swift
index 34a5bad..90117eb 100644
--- a/node_modules/react-native-change-icon/ios/ChangeIcon.swift
+++ b/node_modules/react-native-change-icon/ios/ChangeIcon.swift
@@ -20,7 +20,13 @@ class ChangeIcon: NSObject {
                 return
             }
             resolve(true)
-            UIApplication.shared.setAlternateIconName(iconName)
+            UIApplication.shared.setAlternateIconName(iconName) { error in
+                if let error = error {
+                     print(error.localizedDescription)
+                } else {
+                     print("Success!")
+                }
+             }
         }
     }
 }
