diff --git a/node_modules/react-native-mmkv/lib/module/hooks.js b/node_modules/react-native-mmkv/lib/module/hooks.js
index 3341a0a..813e9d5 100644
--- a/node_modules/react-native-mmkv/lib/module/hooks.js
+++ b/node_modules/react-native-mmkv/lib/module/hooks.js
@@ -66,7 +66,7 @@ function createMMKVHook(getter) {
     useEffect(() => {
       const listener = mmkv.addOnValueChangedListener(changedKey => {
         if (changedKey === key) {
-          setValue(getter(mmkv, key));
+          setImmediate(() => setValue(getter(mmkv, key)));
         }
       });
       return () => listener.remove();
@@ -146,6 +146,13 @@ export function useMMKVObject(key, instance) {
     if (v == null) {
       // Clear the Value
       setString(undefined);
+    } else if (typeof v === 'function') {
+      setString(prev => {
+        if (prev === undefined) {
+          return JSON.stringify(v(undefined))
+        }
+        return JSON.stringify(v(JSON.parse(prev)))
+      });
     } else {
       // Store the Object as a serialized Value
       setString(JSON.stringify(v));
diff --git a/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts b/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
index b4a07cd..4a8856d 100644
--- a/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
+++ b/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
@@ -64,7 +64,7 @@ export declare const useMMKVBuffer: (key: string, instance?: MMKV | undefined) =
  * const [user, setUser] = useMMKVObject<User>("user")
  * ```
  */
-export declare function useMMKVObject<T>(key: string, instance?: MMKV): [value: T | undefined, setValue: (value: T | undefined) => void];
+export declare function useMMKVObject<T>(key: string, instance?: MMKV): [value: T | undefined, setValue: (value: T | ((current: T) => T | undefined) | undefined) => void];
 /**
  * Listen for changes in the given MMKV storage instance.
  * If no instance is passed, the default instance will be used.
