diff --git a/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButton.swift b/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButton.swift
index 50ac183..dea3cea 100644
--- a/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButton.swift
+++ b/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButton.swift
@@ -42,6 +42,7 @@ class RNIContextMenuButton: UIButton {
 
   // MARK: - RN Exported Props
   // -------------------------
+  @objc var menuAlignmentOverride: NSString?
 
   @objc var menuConfig: NSDictionary? {
     didSet {
@@ -279,3 +280,21 @@ extension RNIContextMenuButton: RNIContextMenu {
     childVC.removeFromParent();
   };
 };
+
+// MARK: - Custom alignment
+// ------------------------
+
+@available(iOS 14, *)
+extension RNIContextMenuButton {
+  override func menuAttachmentPoint(for configuration: UIContextMenuConfiguration) -> CGPoint {
+    let original = super.menuAttachmentPoint(for: configuration)
+    
+    if (self.menuAlignmentOverride == "left") {
+      return CGPoint(x: 0, y: original.y)
+    } else if (self.menuAlignmentOverride == "right") {
+      return CGPoint(x: self.bounds.width, y: original.y)
+    }
+    
+    return original
+  }
+}
diff --git a/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButtonManager.m b/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButtonManager.m
index 511ee58..5bc275f 100644
--- a/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButtonManager.m
+++ b/node_modules/react-native-ios-context-menu/ios/src/ReactNative/RNIContextMenuButton/RNIContextMenuButtonManager.m
@@ -34,6 +34,7 @@ @interface RCT_EXTERN_MODULE(RNIContextMenuButtonManager, RCTViewManager)
 RCT_EXPORT_VIEW_PROPERTY(enableContextMenu   , BOOL);
 RCT_EXPORT_VIEW_PROPERTY(isMenuPrimaryAction , BOOL);
 RCT_EXPORT_VIEW_PROPERTY(isContextMenuEnabled, BOOL);
+RCT_EXPORT_VIEW_PROPERTY(menuAlignmentOverride, NSString);
 
 // MARK: -  View Manager Commands
 // ------------------------------
diff --git a/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButton.d.ts b/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButton.d.ts
index 37719d5..d16c85b 100644
--- a/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButton.d.ts
+++ b/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButton.d.ts
@@ -8,6 +8,7 @@ export declare class ContextMenuButton extends React.PureComponent<ContextMenuBu
         enableContextMenu: boolean;
         useActionSheetFallback: boolean;
         menuConfig: import("../..").MenuConfig | undefined;
+        menuAlignmentOverride: 'left' | 'right' | undefined;
         isMenuPrimaryAction: boolean | undefined;
         onMenuWillShow: OnMenuWillShowEvent | undefined;
         onMenuWillHide: OnMenuWillHideEvent | undefined;
diff --git a/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButtonTypes.d.ts b/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButtonTypes.d.ts
index 02f7fb9..3d3e395 100644
--- a/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButtonTypes.d.ts
+++ b/node_modules/react-native-ios-context-menu/lib/typescript/components/ContextMenuButton/ContextMenuButtonTypes.d.ts
@@ -1,6 +1,6 @@
 import type { ViewProps } from 'react-native';
 import type { RNIContextMenuButtonBaseProps } from '../../native_components/RNIContextMenuButton';
-export declare type ContextMenuButtonBaseProps = Pick<RNIContextMenuButtonBaseProps, 'enableContextMenu' | 'isMenuPrimaryAction' | 'menuConfig' | 'onMenuWillShow' | 'onMenuWillHide' | 'onMenuWillCancel' | 'onMenuDidShow' | 'onMenuDidHide' | 'onMenuDidCancel' | 'onPressMenuItem'> & {
+export declare type ContextMenuButtonBaseProps = Pick<RNIContextMenuButtonBaseProps, 'enableContextMenu' | 'menuAlignmentOverride' | 'isMenuPrimaryAction' | 'menuConfig' | 'onMenuWillShow' | 'onMenuWillHide' | 'onMenuWillCancel' | 'onMenuDidShow' | 'onMenuDidHide' | 'onMenuDidCancel' | 'onPressMenuItem'> & {
     useActionSheetFallback?: boolean;
 };
 export declare type ContextMenuButtonProps = ViewProps & ContextMenuButtonBaseProps;
diff --git a/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuButton.d.ts b/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuButton.d.ts
index ad9f406..d5affc7 100644
--- a/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuButton.d.ts
+++ b/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuButton.d.ts
@@ -1,6 +1,6 @@
 import { HostComponent, ViewProps } from 'react-native';
 import type { RNIContextMenuViewProps } from './RNIContextMenuView';
-export declare type RNIContextMenuButtonBaseProps = Pick<RNIContextMenuViewProps, 'menuConfig' | 'onMenuWillShow' | 'onMenuWillHide' | 'onMenuWillCancel' | 'onMenuDidShow' | 'onMenuDidHide' | 'onMenuDidCancel' | 'onPressMenuItem'> & {
+export declare type RNIContextMenuButtonBaseProps = Pick<RNIContextMenuViewProps, 'menuConfig' | 'menuAlignmentOverride' | 'onMenuWillShow' | 'onMenuWillHide' | 'onMenuWillCancel' | 'onMenuDidShow' | 'onMenuDidHide' | 'onMenuDidCancel' | 'onPressMenuItem'> & {
     enableContextMenu?: boolean;
     isMenuPrimaryAction?: boolean;
 };
diff --git a/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuView.d.ts b/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuView.d.ts
index 8fc2acd..09c7b33 100644
--- a/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuView.d.ts
+++ b/node_modules/react-native-ios-context-menu/lib/typescript/native_components/RNIContextMenuView.d.ts
@@ -5,6 +5,7 @@ import type { MenuAuxiliaryPreviewConfig } from '../types/MenuAuxiliaryPreviewCo
 import type { OnMenuWillShowEvent, OnMenuWillHideEvent, OnMenuDidShowEvent, OnMenuDidHideEvent, OnMenuWillCancelEvent, OnMenuDidCancelEvent, OnMenuWillCreateEvent, OnPressMenuItemEvent, OnPressMenuPreviewEvent } from '../types/MenuEvents';
 export declare type RNIContextMenuViewProps = ViewProps & {
     menuConfig?: MenuConfig;
+    menuAlignmentOverride?: 'left' | 'right';
     previewConfig?: MenuPreviewConfig;
     auxiliaryPreviewConfig?: MenuAuxiliaryPreviewConfig;
     /**
diff --git a/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButton.tsx b/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButton.tsx
index e545806..4fbb9ac 100644
--- a/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButton.tsx
+++ b/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButton.tsx
@@ -1,5 +1,6 @@
 import React from 'react';
-import { StyleSheet, View, TouchableOpacity, UIManager, findNodeHandle } from 'react-native';
+import { StyleSheet, View, UIManager, findNodeHandle } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler'
 
 import { RNIContextMenuButton, RNIContextMenuButtonBaseProps, RNIContextMenuButtonCommands } from '../../native_components/RNIContextMenuButton';
 
@@ -211,7 +212,9 @@ export class ContextMenuButton extends React.PureComponent<ContextMenuButtonProp
       ): shouldUseActionSheetFallback? (
         // C - Use 'ActionSheet' Fallback
         <TouchableOpacity 
-          onLongPress={this._handleOnLongPress}
+          {...({
+            [props.isMenuPrimaryAction ? 'onPress' : 'onLongPress']:this._handleOnLongPress
+          })}
           activeOpacity={0.8}
           {...props.viewProps}
         >
diff --git a/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButtonTypes.tsx b/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButtonTypes.tsx
index da57e78..bd122c3 100644
--- a/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButtonTypes.tsx
+++ b/node_modules/react-native-ios-context-menu/src/components/ContextMenuButton/ContextMenuButtonTypes.tsx
@@ -7,6 +7,7 @@ export type ContextMenuButtonBaseProps = Pick<RNIContextMenuButtonBaseProps,
   | 'enableContextMenu'
   | 'isMenuPrimaryAction'
   | 'menuConfig'
+  | 'menuAlignmentOverride'
   // Lifecycle Events
   | 'onMenuWillShow'
   | 'onMenuWillHide'
diff --git a/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuButton.ts b/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuButton.ts
index a609a8a..0acf10b 100644
--- a/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuButton.ts
+++ b/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuButton.ts
@@ -4,6 +4,7 @@ import type { RNIContextMenuViewProps } from './RNIContextMenuView';
 
 export type RNIContextMenuButtonBaseProps = Pick<RNIContextMenuViewProps,
   | 'menuConfig'
+  | 'menuAlignmentOverride'
   // Lifecycle Events
   | 'onMenuWillShow'
   | 'onMenuWillHide'
diff --git a/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuView.ts b/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuView.ts
index a54e54a..361f9e4 100644
--- a/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuView.ts
+++ b/node_modules/react-native-ios-context-menu/src/native_components/RNIContextMenuView.ts
@@ -13,6 +13,7 @@ export type RNIContextMenuViewProps = ViewProps & {
   // -----------
 
   menuConfig?: MenuConfig; 
+  menuAlignmentOverride?: 'left' | 'right';
   previewConfig?: MenuPreviewConfig;
   auxiliaryPreviewConfig?: MenuAuxiliaryPreviewConfig;
 
