diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandlerManager.h b/node_modules/react-native-gesture-handler/ios/RNGestureHandlerManager.h
index a94a0fd..3c2c4a6 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandlerManager.h
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandlerManager.h
@@ -2,6 +2,7 @@
 
 #import <React/RCTBridgeModule.h>
 #import <React/RCTUIManager.h>
+#import <React/RCTEventDispatcher.h>
 
 @interface RNGestureHandlerManager : NSObject
 
