diff --git a/node_modules/react-native-radial-gradient/android/src/main/java/com/surajit/rnrg/GradientView.java b/node_modules/react-native-radial-gradient/android/src/main/java/com/surajit/rnrg/GradientView.java
index f3c1d49..0b3a0a8 100644
--- a/node_modules/react-native-radial-gradient/android/src/main/java/com/surajit/rnrg/GradientView.java
+++ b/node_modules/react-native-radial-gradient/android/src/main/java/com/surajit/rnrg/GradientView.java
@@ -7,15 +7,14 @@ import android.graphics.Paint;
 import android.graphics.RadialGradient;
 import android.graphics.RectF;
 import android.graphics.Shader;
-import android.support.annotation.Nullable;
-import android.support.v4.graphics.ColorUtils;
+import androidx.annotation.Nullable;
+import androidx.core.graphics.ColorUtils;
 import android.util.AttributeSet;
 import android.view.View;

 import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.uimanager.PixelUtil;

-import java.text.DecimalFormat;

 /**
  * Created by Surajit Sarkar on 22/12/17.