diff --git a/node_modules/react-native-text-input-mask/ios/RNTextInputMask.m b/node_modules/react-native-text-input-mask/ios/RNTextInputMask.m
index 847a592..fb6fa42 100644
--- a/node_modules/react-native-text-input-mask/ios/RNTextInputMask.m
+++ b/node_modules/react-native-text-input-mask/ios/RNTextInputMask.m
@@ -12,6 +12,7 @@ @implementation RNTextInputMask {
 }
 
 @synthesize bridge = _bridge;
+@synthesize methodQueue = _methodQueue;
 
 RCT_EXPORT_MODULE();
 
