diff --git a/node_modules/react-native-cool-modals/createNativeStackNavigator.js b/node_modules/react-native-cool-modals/createNativeStackNavigator.js
index 334ef99..e3695c7 100644
--- a/node_modules/react-native-cool-modals/createNativeStackNavigator.js
+++ b/node_modules/react-native-cool-modals/createNativeStackNavigator.js
@@ -61,6 +61,7 @@ class StackView extends React.Component {

     let stackPresentation = 'push';
     const {
+      dismissable,
       customStack,
       topOffset,
       showDragIndicator,
@@ -98,6 +99,7 @@ class StackView extends React.Component {
     const { screenProps } = this.props;
     return (
       <Screen
+        dismissable={dismissable}
         onWillDismiss={onWillDismiss}
         allowsDragToDismiss={allowsDragToDismiss}
         allowsTapToDismiss={allowsTapToDismiss}
diff --git a/node_modules/react-native-cool-modals/ios/RNCMScreen.h b/node_modules/react-native-cool-modals/ios/RNCMScreen.h
index 8648e95..fe585da 100644
--- a/node_modules/react-native-cool-modals/ios/RNCMScreen.h
+++ b/node_modules/react-native-cool-modals/ios/RNCMScreen.h
@@ -50,6 +50,7 @@ typedef NS_ENUM(NSInteger, RNSScreenStackAnimation) {
 @property (nonatomic) BOOL active;
 @property (nonatomic) BOOL customStack;
 @property (nonatomic) BOOL gestureEnabled;
+@property (nonatomic) BOOL dismissable;
 @property (nonatomic) BOOL showDragIndicator;
 @property (nonatomic) NSNumber* topOffset;
 @property (nonatomic) NSNumber* cornerRadius;
diff --git a/node_modules/react-native-cool-modals/ios/RNCMScreen.m b/node_modules/react-native-cool-modals/ios/RNCMScreen.m
index 7c27c15..afdf6e5 100644
--- a/node_modules/react-native-cool-modals/ios/RNCMScreen.m
+++ b/node_modules/react-native-cool-modals/ios/RNCMScreen.m
@@ -45,7 +45,7 @@ - (instancetype)initWithBridge:(RCTBridge *)bridge
     _startFromShortForm = false;
     _modalBackgroundColor = [[UIColor alloc] initWithRed:0.0f green:0.0f blue:0.0f alpha:1];
     _backgroundOpacity = [[NSNumber alloc] initWithDouble:0.7];
-
+    _dismissable = YES;
   }

   return self;
@@ -375,6 +375,7 @@ @implementation RNCMScreenManager
 RCT_EXPORT_VIEW_PROPERTY(gestureEnabled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(showDragIndicator, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(customStack, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(dismissable, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(topOffset, NSNumber)
 RCT_EXPORT_VIEW_PROPERTY(cornerRadius, NSNumber)
 RCT_EXPORT_VIEW_PROPERTY(stackPresentation, RNSScreenStackPresentation)
diff --git a/node_modules/react-native-cool-modals/ios/RNCMScreenStack.m b/node_modules/react-native-cool-modals/ios/RNCMScreenStack.m
index b397efb..b9194e9 100644
--- a/node_modules/react-native-cool-modals/ios/RNCMScreenStack.m
+++ b/node_modules/react-native-cool-modals/ios/RNCMScreenStack.m
@@ -315,10 +315,10 @@ - (void)setModalViewControllers:(NSArray<UIViewController *> *)controllers
   }
   if (presentedViewController != nil
       && ([_presentedModals containsObject:presentedViewController] )) {
-    [changeRootController
-     dismissViewControllerAnimated:(changeRootIndex == controllers.count)
-     completion:finish];
-     //[RNCMScreenStackView dismissViewControllerWrapper: changeRootController animated:(changeRootIndex == controllers.count) completion:finish];
+//    [changeRootController
+//     dismissViewControllerAnimated:(changeRootIndex == controllers.count)
+//     completion:finish];
+     [RNCMScreenStackView dismissViewControllerWrapper: changeRootController animated:(changeRootIndex == controllers.count) completion:finish];
   } else {
     finish();
   }
diff --git a/node_modules/react-native-cool-modals/ios/UIViewController+slack.swift b/node_modules/react-native-cool-modals/ios/UIViewController+slack.swift
index b21a043..b455899 100644
--- a/node_modules/react-native-cool-modals/ios/UIViewController+slack.swift
+++ b/node_modules/react-native-cool-modals/ios/UIViewController+slack.swift
@@ -125,7 +125,10 @@ class PanModalViewController: UIViewController, PanModalPresentable, UILayoutSup
   }

   func shouldRespond(to panModalGestureRecognizer: UIPanGestureRecognizer) -> Bool {
-    return !hiding
+    if (hiding) {
+      return false
+    }
+    return self.config?.value(forKey: "dismissable") as! Bool
   }

   var allowsDragToDismiss: Bool {
diff --git a/node_modules/react-native-cool-modals/native-stack/views/NativeStackView.tsx b/node_modules/react-native-cool-modals/native-stack/views/NativeStackView.tsx
index c2f2e2b..7f6f332 100644
--- a/node_modules/react-native-cool-modals/native-stack/views/NativeStackView.tsx
+++ b/node_modules/react-native-cool-modals/native-stack/views/NativeStackView.tsx
@@ -43,6 +43,7 @@ export default function NativeStackView({
           } = options;

           const {
+            dismissable,
             customStack,
             topOffset,
             showDragIndicator,
@@ -62,6 +63,7 @@ export default function NativeStackView({

           return (
               <Screen
+                  dismissable={dismissable}
                   customStack={customStack}
                   topOffset={topOffset}
                   showDragIndicator={showDragIndicator}
