diff --git a/node_modules/react-native-svg/apple/Elements/RNSVGGroup.mm b/node_modules/react-native-svg/apple/Elements/RNSVGGroup.mm
index 17138e6..45c6760 100644
--- a/node_modules/react-native-svg/apple/Elements/RNSVGGroup.mm
+++ b/node_modules/react-native-svg/apple/Elements/RNSVGGroup.mm
@@ -189,7 +189,6 @@ using namespace facebook::react;
   [self traverseSubviews:^(RNSVGNode *node) {
     if ([node isKindOfClass:[RNSVGNode class]] && ![node isKindOfClass:[RNSVGMask class]]) {
       CGAffineTransform transform = CGAffineTransformConcat(node.matrix, node.transforms);
-      CGPathAddPath(path, &transform, [node getPath:context]);
       CGPathAddPath(path, &transform, [node markerPath]);
       node.dirty = false;
     }
diff --git a/node_modules/react-native-svg/apple/RNSVGRenderable.mm b/node_modules/react-native-svg/apple/RNSVGRenderable.mm
index d69b974..59ac4d5 100644
--- a/node_modules/react-native-svg/apple/RNSVGRenderable.mm
+++ b/node_modules/react-native-svg/apple/RNSVGRenderable.mm
@@ -431,6 +431,7 @@ UInt32 saturate(CGFloat value)
   if (_vectorEffect == kRNSVGVectorEffectNonScalingStroke) {
     path = CGPathCreateCopyByTransformingPath(path, &svgToClientTransform);
     CGContextConcatCTM(context, CGAffineTransformInvert(svgToClientTransform));
+    CFAutorelease(path);
   }
 
   CGAffineTransform vbmatrix = self.svgView.getViewBoxTransform;
