diff --git a/node_modules/react-native-mmkv/lib/commonjs/hooks.js b/node_modules/react-native-mmkv/lib/commonjs/hooks.js
index 9382032..c1d8ffc 100644
--- a/node_modules/react-native-mmkv/lib/commonjs/hooks.js
+++ b/node_modules/react-native-mmkv/lib/commonjs/hooks.js
@@ -155,8 +155,14 @@ function useMMKVObject(key, instance) {
     if (v == null) {
       // Clear the Value
       setString(undefined);
+    } else if (typeof v === 'function') {
+      setString(prev => {
+        if (prev === undefined) {
+          return JSON.stringify(v(undefined))
+       }
+        return JSON.stringify(v(JSON.parse(prev)))
+      });
     } else {
-      // Store the Object as a serialized Value
       setString(JSON.stringify(v));
     }
   }, [setString]);
diff --git a/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts b/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
index 54c6ae9..6d992a7 100644
--- a/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
+++ b/node_modules/react-native-mmkv/lib/typescript/hooks.d.ts
@@ -53,7 +53,7 @@ export declare const useMMKVBoolean: (key: string, instance?: MMKV | undefined)
  * const [user, setUser] = useMMKVObject<User>("user")
  * ```
  */
-export declare function useMMKVObject<T>(key: string, instance?: MMKV): [value: T | undefined, setValue: (value: T | undefined) => void];
+export declare function useMMKVObject<T>(key: string, instance?: MMKV): [value: T | undefined, setValue: (value: T | ((current: T) => T | undefined) | undefined) => void];
 /**
  * Listen for changes in the given MMKV storage instance.
  * If no instance is passed, the default instance will be used.
