diff --git a/node_modules/recyclerlistview/dist/reactnative/core/sticky/StickyHeader.js b/node_modules/recyclerlistview/dist/reactnative/core/sticky/StickyHeader.js
index 7074f36..153ddb4 100644
--- a/node_modules/recyclerlistview/dist/reactnative/core/sticky/StickyHeader.js
+++ b/node_modules/recyclerlistview/dist/reactnative/core/sticky/StickyHeader.js
@@ -60,7 +60,7 @@ var StickyHeader = /** @class */ (function (_super) {
         return offsetY;
     };
     StickyHeader.prototype.hasReachedBoundary = function (offsetY, distanceFromWindow, _windowBound) {
-        return offsetY <= distanceFromWindow;
+        return offsetY < distanceFromWindow;
     };
     return StickyHeader;
 }(StickyObject_1.default));
