diff --git a/node_modules/playroom/lib/start.js b/node_modules/playroom/lib/start.js
index 0d9d5dd..e19d147 100644
--- a/node_modules/playroom/lib/start.js
+++ b/node_modules/playroom/lib/start.js
@@ -37,6 +37,9 @@ module.exports = async (config, callback) => {
       // Added to prevent Webpack HMR from breaking when iframeSandbox option is used
       // See: https://github.com/webpack/webpack-dev-server/issues/1604
       allowedHosts: 'all',
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+      },
     };
 
     const compiler = webpack(webpackConfig);
